diff -ruN psi-pedrito-2006-02-03./src/adhoc_fileserver.h psi-pedrito-2006-02-03/src/adhoc_fileserver.h
--- psi-pedrito-2006-02-03./src/adhoc_fileserver.h	2006-01-18 00:48:36.000000000 +0100
+++ psi-pedrito-2006-02-03/src/adhoc_fileserver.h	2006-07-18 11:41:20.797973000 +0200
@@ -30,6 +30,10 @@
 	AHFileServer(AHCServerManager* m) : AHCommandServer(m) {
 		this->m = m;
 	}
+	virtual ~AHFileServer()
+	{
+	}
+
 	virtual QString node() const 
 		{ return QString("http://psi.affinix.com/commands/files"); }
 	virtual bool isAllowed(const Jid&) const;
diff -ruN psi-pedrito-2006-02-03,/src/adhoc.h psi-pedrito-2006-02-03/src/adhoc.h
--- psi-pedrito-2006-02-03,/src/adhoc.h	2005-12-12 01:28:13.000000000 +0100
+++ psi-pedrito-2006-02-03/src/adhoc.h	2006-07-18 13:35:36.774444750 +0200
@@ -134,6 +134,9 @@
 {
 public:
 	AHCommandServer(AHCServerManager*);
+	virtual ~AHCommandServer()
+	{
+	}
 
 	virtual QString name() const = 0; 
 	virtual QString node() const = 0; 
diff -ruN psi-pedrito-2006-02-03./src/adhoc_fileserver.h psi-pedrito-2006-02-03/src/adhoc_fileserver.h
--- psi-pedrito-2006-02-03./src/adhoc_fileserver.h	2006-07-18 13:36:09.000000000 +0200
+++ psi-pedrito-2006-02-03/src/adhoc_fileserver.h	2006-07-18 13:48:29.682748500 +0200
@@ -40,7 +40,7 @@
 	virtual QString name() const { return QString(QObject::tr("Get list of shared files")); }
 	virtual AHCommand execute(const AHCommand& c, const Jid&);
 
-	bool AHFileServer::ACL_check(QStringList ACL, const Jid& j) const;
+	bool ACL_check(QStringList ACL, const Jid& j) const;
 
 private:
 	AHCServerManager* m;
