diff -urN psi-0.10.orig/src/chatdlg.cpp psi-0.10/src/chatdlg.cpp
--- psi-0.10.orig/src/chatdlg.cpp	2006-01-19 18:38:13.000000000 +0100
+++ psi-0.10/src/chatdlg.cpp	2006-01-19 18:41:28.796362000 +0100
@@ -692,6 +692,12 @@
 	setIcon(IconsetFactory::icon("psi/start-chat"));
 #endif
 
+	d->log->setPaletteBackgroundColor(option.color[cChatBG]);
+	d->mle->setPaletteBackgroundColor(option.color[cChatBG]);
+
+	d->log->setPaletteForegroundColor(option.color[cChatMessage]);
+	d->mle->setPaletteForegroundColor(option.color[cChatMessage]);
+
 	/*QBrush brush;
 	brush.setPixmap( QPixmap( option.chatBgImage ) );
 	d->log->setPaper(brush);
@@ -933,7 +939,7 @@
 		y = 0;
 	bool atBottom = (d->log->contentsY() < y - 32) ? false: true;
 
-	d->log->append(QString("<font color=\"#00A000\">[%1]").arg(timestr) + QString(" *** %1</font>").arg(str));
+	d->log->append(QString("<font color=\"" + option.color[cChatSys].name() + "\">(%1)").arg(timestr) + QString(" *** %1</font>").arg(str));
 	if(atBottom)
 		deferredScroll();
 }
@@ -944,14 +950,14 @@
 
 	if(local) {
 		who = d->pa->nick();
-		color = "#FF0000";
+		color = option.color[cChatTo].name();
 	}
 	else {
 		who = d->dispNick;
-		color = "#0000FF";
+		color = option.color[cChatFrom].name();
 	}
 	if(m.spooled())
-		color = "#008000";
+		color = option.color[cChatSpool].name();
 
 	// figure out the encryption state
 	bool encChanged = false;
@@ -980,8 +986,12 @@
 	}
 
 	QString timestr;
-	QDateTime time = m.timeStamp();
-	timestr = time.time().toString(LocalDate);
+	if(option.showTimeStamps)
+	{
+		QDateTime time = m.timeStamp();
+		//timestr.sprintf("(%02d:%02d:%02d) ", time.time().hour(), time.time().minute(), time.time().second());
+		timestr = QString("(") + time.time().toString(LocalDate) + ") ";
+	}
 
 	int y = d->log->contentsHeight() - d->log->visibleHeight();
 	if(y < 0)
@@ -1003,16 +1013,19 @@
 	if(option.useEmoticons)
 		txt = emoticonify(txt);
 
+	if(!emote)
+		txt = QString("<font color=\"%1\">").arg(option.color[cChatMessage].name()) + txt +"</font>";
+
 	who = expandEntities(who);
 
 	if(emote) {
-		d->log->append(QString("<font color=\"%1\">").arg(color) + QString("[%1]").arg(timestr) + QString(" *%1 ").arg(who) + txt + "</font>");
+		d->log->append(QString("<font color=\"%1\"><i>").arg(color) + timestr + QString("**%1 ").arg(who) + txt + QString("</i></font>"));
 	}
 	else {
 		if(option.chatSays)
-			d->log->append(QString("<font color=\"%1\">").arg(color) + QString("[%1] ").arg(timestr) + tr("%1 says:").arg(who) + "</font><br>" + txt);
+			d->log->append(QString("<font color=\"%1\">").arg(color) + timestr + tr("%1 says:").arg(who) + "</font><br>" + txt);
 		else
-			d->log->append(QString("<font color=\"%1\">").arg(color) + QString("[%1] &lt;").arg(timestr) + who + QString("&gt;</font> ") + txt);
+			d->log->append(QString("<font color=\"%1\">").arg(color) + timestr + who + QString(":</font> ") + txt);
 	}
 	if(!m.subject().isEmpty()) {
 		d->log->append(QString("<b>") + tr("Subject:") + "</b> " + QString("%1").arg(expandEntities(m.subject())));
diff -urN psi-0.10.orig/src/common.h psi-0.10/src/common.h
--- psi-0.10.orig/src/common.h	2006-01-19 18:38:13.000000000 +0100
+++ psi-0.10/src/common.h	2006-01-20 03:12:50.212032000 +0100
@@ -69,14 +69,14 @@
 extern uint psi_dialog_flags;
 
 // options stuff
-enum { cOnline, cOffline, cAway, cDND, cProfileFore, cProfileBack, cGroupFore, cGroupBack, cListBack, cAnimFront, cAnimBack };
+enum { cOnline, cOffline, cAway, cDND, cProfileFore, cProfileBack, cGroupFore, cGroupBack, cListBack, cAnimFront, cAnimBack, cChatBG, cChatMessage, cChatTo, cChatFrom, cChatSys, cChatSpool, colorLast };
 enum { fRoster, fMessage, fChat, fPopup };
 enum { eMessage, eChat1, eChat2, eHeadline, eSystem, eOnline, eOffline, eSend, eIncomingFT, eFTComplete };
 enum { dcClose, dcHour, dcDay, dcNever };
 
 struct Options
 {
-	QColor color[11];
+	QColor color[colorLast];
 	QString font[4];
 	int smallFontSize; // do not modify or save/load this value! it is calculated at run time!
 	int alertStyle;
@@ -123,7 +123,7 @@
 	QString customBrowser, customMailer;
 
 	bool ignoreHeadline, ignoreNonRoster, excludeGroupChatsFromIgnore, scrollTo, keepSizes, useEmoticons, alertOpenChats;
-	bool raiseChatWindow, showSubjects, showCounter, chatSays, chatSoftReturn, showGroupCounts;
+	bool raiseChatWindow, showSubjects, showCounter, chatSays, showTimeStamps, showGroupCounts, chatSoftReturn;
 
 	QSize sizeEventDlg, sizeChatDlg, sizeTabDlg;
 	bool jidComplete, grabUrls, noGCSound;
diff -urN psi-0.10.orig/src/options/opt_appearance.cpp psi-0.10/src/options/opt_appearance.cpp
--- psi-0.10.orig/src/options/opt_appearance.cpp	2006-01-19 18:38:13.000000000 +0100
+++ psi-0.10/src/options/opt_appearance.cpp	2006-01-20 03:10:13.512032000 +0100
@@ -210,6 +210,12 @@
 	bg_color->insert(d->pb_cListBack, 8);
 	bg_color->insert(d->pb_cAnimFront, 9);
 	bg_color->insert(d->pb_cAnimBack, 10);
+	bg_color->insert(d->pb_cChatBG, 11);
+	bg_color->insert(d->pb_cChatMessage, 12);
+	bg_color->insert(d->pb_cChatTo, 13);
+	bg_color->insert(d->pb_cChatFrom, 14);
+	bg_color->insert(d->pb_cChatSys, 15);
+	bg_color->insert(d->pb_cChatSpool, 16);
 	connect(bg_color, SIGNAL(clicked(int)), SLOT(chooseColor(int)));
 
 	QString s = tr("Specifies the text color for a contact name in the main window when that user is \"%1\".");
@@ -231,6 +237,18 @@
 		tr("Specifies the foreground animation color for nicks."));
 	QWhatsThis::add(d->pb_cAnimBack,
 		tr("Specifies the background animation color for nicks."));
+	QWhatsThis::add(d->pb_cChatBG,
+		tr("Specifies the color for the chat window."));
+	QWhatsThis::add(d->pb_cChatMessage,
+		tr("Specifies the color for chat message text."));
+	QWhatsThis::add(d->pb_cChatTo,
+		tr("Specifies the color for sent chat messages."));
+	QWhatsThis::add(d->pb_cChatFrom,
+		tr("Specifies the color for recieved chat messages."));
+	QWhatsThis::add(d->pb_cChatSys,
+		tr("Specifies the color for system chat messages."));
+	QWhatsThis::add(d->pb_cChatSpool,
+		tr("Specifies the color for spooled chat messages."));
 	QWhatsThis::add(d->ck_statusIndicator,
 		tr("Displays an indicator by the contact name if the contact entered a descriptive text status line."));
 
@@ -254,7 +272,7 @@
 	for (n = 0; n < 4; ++n)
 		opt->font[n] = le_font[n]->fontName();
 
-	for (n = 0; n < 11; ++n)
+	for (n = 0; n < 17; ++n)
 		opt->color[n] = o->color[n];
 }
 
@@ -271,7 +289,7 @@
 	for (n = 0; n < 4; ++n)
 		le_font[n]->setFont(opt->font[n]);
 
-	for (n = 0; n < 11; ++n) {
+	for (n = 0; n < 17; ++n) {
 		o->color[n] = opt->color[n];
 		((QPushButton *)bg_color->find(n))->setPixmap(name2color(opt->color[n].name()));
 	}
diff -urN psi-0.10.orig/src/options/opt_appearance-ui.ui psi-0.10/src/options/opt_appearance-ui.ui
--- psi-0.10.orig/src/options/opt_appearance-ui.ui	2006-01-19 18:38:13.000000000 +0100
+++ psi-0.10/src/options/opt_appearance-ui.ui	2006-01-20 03:08:46.082032000 +0100
@@ -36,25 +36,47 @@
                 <property name="spacing">
                     <number>3</number>
                 </property>
-                <widget class="QLabel" row="5" column="0">
+                <widget class="QToolButton" row="0" column="4">
                     <property name="name">
-                        <cstring>text_1</cstring>
+                        <cstring>pb_cAnimFront</cstring>
+                    </property>
+                    <property name="sizePolicy">
+                        <sizepolicy>
+                            <hsizetype>0</hsizetype>
+                            <vsizetype>0</vsizetype>
+                            <horstretch>0</horstretch>
+                            <verstretch>0</verstretch>
+                        </sizepolicy>
+                    </property>
+                    <property name="maximumSize">
+                        <size>
+                            <width>20</width>
+                            <height>20</height>
+                        </size>
                     </property>
                     <property name="text">
-                        <string>Account Heading Background</string>
+                        <string></string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="7" column="0">
+                <widget class="QLabel" row="4" column="3">
                     <property name="name">
-                        <cstring>text</cstring>
+                        <cstring>TextLabel1_2_5_2</cstring>
                     </property>
                     <property name="text">
-                        <string>Group Heading Background</string>
+                        <string>Sent Chat Messages</string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="8" column="1">
+                <widget class="QLabel" row="6" column="3">
                     <property name="name">
-                        <cstring>pb_cListBack</cstring>
+                        <cstring>text4_2</cstring>
+                    </property>
+                    <property name="text">
+                        <string>System Chat Messages</string>
+                    </property>
+                </widget>
+                <widget class="QToolButton" row="7" column="4">
+                    <property name="name">
+                        <cstring>pb_cChatSpool</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -74,7 +96,7 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="10" column="1">
+                <widget class="QToolButton" row="1" column="4">
                     <property name="name">
                         <cstring>pb_cAnimBack</cstring>
                     </property>
@@ -96,33 +118,79 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="1" column="0">
+                <widget class="QToolButton" row="2" column="4">
                     <property name="name">
-                        <cstring>TextLabel1_2_3</cstring>
+                        <cstring>pb_cChatBG</cstring>
+                    </property>
+                    <property name="sizePolicy">
+                        <sizepolicy>
+                            <hsizetype>0</hsizetype>
+                            <vsizetype>0</vsizetype>
+                            <horstretch>0</horstretch>
+                            <verstretch>0</verstretch>
+                        </sizepolicy>
+                    </property>
+                    <property name="maximumSize">
+                        <size>
+                            <width>20</width>
+                            <height>20</height>
+                        </size>
                     </property>
                     <property name="text">
-                        <string>Offline Contacts</string>
+                        <string></string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="2" column="0">
+                <widget class="QLabel" row="1" column="3">
                     <property name="name">
-                        <cstring>TextLabel1_2</cstring>
+                        <cstring>TextLabel1_2_6_2_2</cstring>
                     </property>
                     <property name="text">
-                        <string>Away Contacts</string>
+                        <string>Nick Animation Background</string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="0" column="0">
+                <widget class="QLabel" row="0" column="3">
                     <property name="name">
-                        <cstring>TextLabel1_2_6</cstring>
+                        <cstring>TextLabel1_2_2</cstring>
                     </property>
                     <property name="text">
-                        <string>Online Contacts</string>
+                        <string>Nick Animation Foreground</string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="5" column="1">
+                <widget class="QLabel" row="5" column="3">
                     <property name="name">
-                        <cstring>pb_cProfileBack</cstring>
+                        <cstring>text_2</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Received Chat Messages</string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="3" column="3">
+                    <property name="name">
+                        <cstring>text2_2</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Chat Message Text</string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="7" column="3">
+                    <property name="name">
+                        <cstring>text3_2</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Spooled Chat Messages</string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="2" column="3">
+                    <property name="name">
+                        <cstring>TextLabel1_2_5_1_2</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Chat Window Background</string>
+                    </property>
+                </widget>
+                <widget class="QToolButton" row="5" column="4">
+                    <property name="name">
+                        <cstring>pb_cChatFrom</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -142,9 +210,9 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="7" column="1">
+                <widget class="QToolButton" row="3" column="4">
                     <property name="name">
-                        <cstring>pb_cGroupBack</cstring>
+                        <cstring>pb_cChatMessage</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -164,9 +232,9 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="3" column="1">
+                <widget class="QToolButton" row="4" column="4">
                     <property name="name">
-                        <cstring>pb_cDND</cstring>
+                        <cstring>pb_cChatTo</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -186,9 +254,9 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="1" column="1">
+                <widget class="QToolButton" row="6" column="4">
                     <property name="name">
-                        <cstring>pb_cOffline</cstring>
+                        <cstring>pb_cChatSys</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -208,9 +276,9 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="4" column="1">
+                <widget class="QToolButton" row="2" column="1">
                     <property name="name">
-                        <cstring>pb_cProfileFore</cstring>
+                        <cstring>pb_cAway</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -230,9 +298,25 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="6" column="1">
+                <widget class="QLabel" row="6" column="0">
                     <property name="name">
-                        <cstring>pb_cGroupFore</cstring>
+                        <cstring>TextLabel1_2_5</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Group Heading Foreground</string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="0" column="0">
+                    <property name="name">
+                        <cstring>TextLabel1_2_6</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Online Contacts</string>
+                    </property>
+                </widget>
+                <widget class="QToolButton" row="0" column="1">
+                    <property name="name">
+                        <cstring>pb_cOnline</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -252,9 +336,9 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="9" column="1">
+                <widget class="QToolButton" row="3" column="1">
                     <property name="name">
-                        <cstring>pb_cAnimFront</cstring>
+                        <cstring>pb_cDND</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -274,9 +358,9 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="0" column="1">
+                <widget class="QToolButton" row="4" column="1">
                     <property name="name">
-                        <cstring>pb_cOnline</cstring>
+                        <cstring>pb_cProfileFore</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -296,36 +380,66 @@
                         <string></string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="9" column="0">
+                <widget class="QLabel" row="3" column="0">
                     <property name="name">
-                        <cstring>text3</cstring>
+                        <cstring>TextLabel1_2_6_2</cstring>
                     </property>
                     <property name="text">
-                        <string>Nick Animation Foreground</string>
+                        <string>DND Contacts</string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="3" column="0">
+                <widget class="QToolButton" row="1" column="1">
                     <property name="name">
-                        <cstring>TextLabel1_2_6_2</cstring>
+                        <cstring>pb_cOffline</cstring>
+                    </property>
+                    <property name="sizePolicy">
+                        <sizepolicy>
+                            <hsizetype>0</hsizetype>
+                            <vsizetype>0</vsizetype>
+                            <horstretch>0</horstretch>
+                            <verstretch>0</verstretch>
+                        </sizepolicy>
+                    </property>
+                    <property name="maximumSize">
+                        <size>
+                            <width>20</width>
+                            <height>20</height>
+                        </size>
                     </property>
                     <property name="text">
-                        <string>DND Contacts</string>
+                        <string></string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="8" column="0">
+                <widget class="QLabel" row="1" column="0">
                     <property name="name">
-                        <cstring>text2</cstring>
+                        <cstring>TextLabel1_2_3</cstring>
                     </property>
                     <property name="text">
-                        <string>Contact List Background</string>
+                        <string>Offline Contacts</string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="10" column="0">
+                <widget class="QLabel" row="2" column="0">
                     <property name="name">
-                        <cstring>text4</cstring>
+                        <cstring>TextLabel1_2</cstring>
                     </property>
                     <property name="text">
-                        <string>Nick Animation Background</string>
+                        <string>Away Contacts</string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="7" column="0">
+                    <property name="name">
+                        <cstring>text</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Group Heading Background</string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="8" column="0">
+                    <property name="name">
+                        <cstring>text2</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Contact List Background</string>
                     </property>
                 </widget>
                 <widget class="QLabel" row="4" column="0">
@@ -336,17 +450,31 @@
                         <string>Account Heading Foreground</string>
                     </property>
                 </widget>
-                <widget class="QLabel" row="6" column="0">
+                <widget class="QToolButton" row="7" column="1">
                     <property name="name">
-                        <cstring>TextLabel1_2_5</cstring>
+                        <cstring>pb_cGroupBack</cstring>
+                    </property>
+                    <property name="sizePolicy">
+                        <sizepolicy>
+                            <hsizetype>0</hsizetype>
+                            <vsizetype>0</vsizetype>
+                            <horstretch>0</horstretch>
+                            <verstretch>0</verstretch>
+                        </sizepolicy>
+                    </property>
+                    <property name="maximumSize">
+                        <size>
+                            <width>20</width>
+                            <height>20</height>
+                        </size>
                     </property>
                     <property name="text">
-                        <string>Group Heading Foreground</string>
+                        <string></string>
                     </property>
                 </widget>
-                <widget class="QToolButton" row="2" column="1">
+                <widget class="QToolButton" row="8" column="1">
                     <property name="name">
-                        <cstring>pb_cAway</cstring>
+                        <cstring>pb_cListBack</cstring>
                     </property>
                     <property name="sizePolicy">
                         <sizepolicy>
@@ -366,6 +494,72 @@
                         <string></string>
                     </property>
                 </widget>
+                <widget class="QToolButton" row="5" column="1">
+                    <property name="name">
+                        <cstring>pb_cProfileBack</cstring>
+                    </property>
+                    <property name="sizePolicy">
+                        <sizepolicy>
+                            <hsizetype>0</hsizetype>
+                            <vsizetype>0</vsizetype>
+                            <horstretch>0</horstretch>
+                            <verstretch>0</verstretch>
+                        </sizepolicy>
+                    </property>
+                    <property name="maximumSize">
+                        <size>
+                            <width>20</width>
+                            <height>20</height>
+                        </size>
+                    </property>
+                    <property name="text">
+                        <string></string>
+                    </property>
+                </widget>
+                <widget class="QToolButton" row="6" column="1">
+                    <property name="name">
+                        <cstring>pb_cGroupFore</cstring>
+                    </property>
+                    <property name="sizePolicy">
+                        <sizepolicy>
+                            <hsizetype>0</hsizetype>
+                            <vsizetype>0</vsizetype>
+                            <horstretch>0</horstretch>
+                            <verstretch>0</verstretch>
+                        </sizepolicy>
+                    </property>
+                    <property name="maximumSize">
+                        <size>
+                            <width>20</width>
+                            <height>20</height>
+                        </size>
+                    </property>
+                    <property name="text">
+                        <string></string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="5" column="0">
+                    <property name="name">
+                        <cstring>text_1</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Account Heading Background</string>
+                    </property>
+                </widget>
+                <widget class="Line" row="0" column="2" rowspan="9" colspan="1">
+                    <property name="name">
+                        <cstring>line1</cstring>
+                    </property>
+                    <property name="frameShape">
+                        <enum>VLine</enum>
+                    </property>
+                    <property name="frameShadow">
+                        <enum>Sunken</enum>
+                    </property>
+                    <property name="orientation">
+                        <enum>Vertical</enum>
+                    </property>
+                </widget>
             </grid>
         </widget>
         <widget class="QCheckBox">
@@ -522,6 +716,12 @@
     <tabstop>pb_cListBack</tabstop>
     <tabstop>pb_cAnimFront</tabstop>
     <tabstop>pb_cAnimBack</tabstop>
+    <tabstop>pb_cChatBG</tabstop>
+    <tabstop>pb_cChatMessage</tabstop>
+    <tabstop>pb_cChatTo</tabstop>
+    <tabstop>pb_cChatFrom</tabstop>
+    <tabstop>pb_cChatSys</tabstop>
+    <tabstop>pb_cChatSpool</tabstop>
     <tabstop>pb_fRoster</tabstop>
     <tabstop>pb_fMessage</tabstop>
     <tabstop>pb_fChat</tabstop>
diff -urN psi-0.10.orig/src/options/opt_chat.cpp psi-0.10/src/options/opt_chat.cpp
--- psi-0.10.orig/src/options/opt_chat.cpp	2006-01-07 22:19:01.000000000 +0100
+++ psi-0.10/src/options/opt_chat.cpp	2006-01-19 18:39:50.046362000 +0100
@@ -62,6 +62,10 @@
 		"<P>Hi</P>"
 		"<P>[01:23:56] YourName says:</P>"
 		"<P>How are you?</P>"));
+	QWhatsThis::add(d->ck_showTimeStamps,
+		tr("<P>Adds a timestamp to chat messages:</P>"
+		"<P>(01:23:45) <MyName> Hi</P>"
+		"<P>(01:23:56) <YourName> How are you?</P>"));
 	QWhatsThis::add(d->ck_chatSoftReturn,
 		tr("<P>When checked, pressing Enter in a chat window will send your message."
 		   "  You must use Shift+Enter in order to create a newline in the chat message."
@@ -100,6 +104,7 @@
 
 	opt->defaultAction   = bg_defAct->id(bg_defAct->selected());
 	opt->chatSays        = d->ck_chatSays->isChecked();
+	opt->showTimeStamps = d->ck_showTimeStamps->isChecked();
 	opt->chatSoftReturn  = d->ck_chatSoftReturn->isChecked();
 	opt->alertOpenChats  = d->ck_alertOpenChats->isChecked();
 	opt->raiseChatWindow = d->ck_raiseChatWindow->isChecked();
@@ -119,6 +124,7 @@
 
 	bg_defAct->setButton( opt->defaultAction );
 	d->ck_chatSays->setChecked( opt->chatSays );
+	d->ck_showTimeStamps->setChecked( opt->showTimeStamps );
 	d->ck_chatSoftReturn->setChecked( opt->chatSoftReturn );
 	d->ck_alertOpenChats->setChecked( opt->alertOpenChats );
 	d->ck_raiseChatWindow->setChecked( opt->raiseChatWindow );
diff -urN psi-0.10.orig/src/options/opt_chat-ui.ui psi-0.10/src/options/opt_chat-ui.ui
--- psi-0.10.orig/src/options/opt_chat-ui.ui	2005-08-22 21:52:28.000000000 +0200
+++ psi-0.10/src/options/opt_chat-ui.ui	2006-01-19 18:39:50.046362000 +0100
@@ -40,6 +40,14 @@
         </widget>
         <widget class="QCheckBox">
             <property name="name">
+                <cstring>ck_showTimeStamps</cstring>
+            </property>
+            <property name="text">
+                <string>Show timestamps in chats</string>
+            </property>
+        </widget>
+        <widget class="QCheckBox">
+            <property name="name">
                 <cstring>ck_alertOpenChats</cstring>
             </property>
             <property name="text">
diff -urN psi-0.10.orig/src/psi_profiles.cpp psi-0.10/src/psi_profiles.cpp
--- psi-0.10.orig/src/psi_profiles.cpp	2006-01-19 18:38:13.000000000 +0100
+++ psi-0.10/src/psi_profiles.cpp	2006-01-20 03:15:19.202032000 +0100
@@ -380,6 +380,7 @@
 	prefs.showSubjects = true;
 	prefs.showCounter = false;
 	prefs.chatSays = false;
+	prefs.showTimeStamps = true;
 	prefs.chatSoftReturn = true;
 	prefs.showGroupCounts = true;
 	prefs.smallChats = false;
@@ -502,6 +503,12 @@
 #else
 		defaultEnableToolbars = true;
 #endif
+	prefs.color[cChatMessage]=QColor("#000000");
+	prefs.color[cChatSys]   = QColor("#444444");
+	prefs.color[cChatTo]    = QColor("#ff4444");
+	prefs.color[cChatFrom]  = QColor("#4444ff");
+	prefs.color[cChatSpool] = QColor("#008000");
+	prefs.color[cChatBG]    = QColor("#e8e8e8");
 
 		// Imitate old Psi button layout by default
 		tb[0].name = QObject::tr("Buttons");
@@ -788,6 +795,7 @@
 			p_misc.appendChild(textTag(doc, "showSubjects", prefs.showSubjects));
 			p_misc.appendChild(textTag(doc, "showCounter", prefs.showCounter));
 			p_misc.appendChild(textTag(doc, "chatSays", prefs.chatSays));
+			p_misc.appendChild(textTag(doc, "showTimeStamps", prefs.showTimeStamps));
 			p_misc.appendChild(textTag(doc, "chatSoftReturn", prefs.chatSoftReturn));
 			p_misc.appendChild(textTag(doc, "showGroupCounts", prefs.showGroupCounts));
 			p_misc.appendChild(textTag(doc, "jidComplete", prefs.jidComplete));
@@ -931,6 +939,12 @@
 			tag.appendChild(textTag(doc, "groupback", prefs.color[cGroupBack].name() ));
 			tag.appendChild(textTag(doc, "animfront", prefs.color[cAnimFront].name() ));
 			tag.appendChild(textTag(doc, "animback", prefs.color[cAnimBack].name() ));
+			tag.appendChild(textTag(doc, "chatbg", prefs.color[cChatBG].name() ));
+			tag.appendChild(textTag(doc, "chatmessage", prefs.color[cChatMessage].name() ));
+			tag.appendChild(textTag(doc, "chatto", prefs.color[cChatTo].name() ));
+			tag.appendChild(textTag(doc, "chatfrom", prefs.color[cChatFrom].name() ));
+			tag.appendChild(textTag(doc, "chatsys", prefs.color[cChatSys].name() ));
+			tag.appendChild(textTag(doc, "chatspool", prefs.color[cChatSpool].name() ));
 		}
 
 		{
@@ -1365,6 +1379,7 @@
 				readBoolEntry(tag, "showGroupCounts", &prefs.showGroupCounts);
 				readBoolEntry(tag, "showCounter", &prefs.showCounter);
 				readBoolEntry(tag, "chatSays", &prefs.chatSays);
+				readBoolEntry(tag, "showTimeStamps", &prefs.showTimeStamps);
 				readBoolEntry(tag, "jidComplete", &prefs.jidComplete);
 				readBoolEntry(tag, "grabUrls", &prefs.grabUrls);
 				readBoolEntry(tag, "smallChats", &prefs.smallChats);
@@ -1515,6 +1530,12 @@
 				readColorEntry(tag, "profileback", &prefs.color[cProfileBack]);
 				readColorEntry(tag, "animfront", &prefs.color[cAnimFront]);
 				readColorEntry(tag, "animback", &prefs.color[cAnimBack]);
+				readColorEntry(tag, "chatbg", &prefs.color[cChatBG]);
+				readColorEntry(tag, "chatmessage", &prefs.color[cChatMessage]);
+				readColorEntry(tag, "chatto", &prefs.color[cChatTo]);
+				readColorEntry(tag, "chatfrom", &prefs.color[cChatFrom]);
+				readColorEntry(tag, "chatsys", &prefs.color[cChatSys]);
+				readColorEntry(tag, "chatspool", &prefs.color[cChatSpool]);
 			}
 
 			tag = findSubTag(p_lnf, "fonts", &found);
