diff -ru psi-0.9.1-test1/src/psi_profiles.cpp psi-0.9.1-test1.orig/src/psi_profiles.cpp
--- psi-0.9.1-test1/src/psi_profiles.cp
+++ psi-0.9.1-test1.orig/src/psi_profiles.cpp
@@ -470,7 +470,7 @@
 	prefs.onevent[eSend]    = g.pathBase + "/sound/send.wav";
 
 	// Added by Kiko 020621: sets up the default certificate store
-	prefs.trustCertStoreDir = g.pathBase + "/certs";
+	prefs.trustCertStoreDir = g.pathHome + "/certs";
 
 	prefs.sizeEventDlg = EventDlg::defaultSize();
 	prefs.sizeChatDlg = ChatDlg::defaultSize();
