--- ./Makefile.orig	Mon Oct  7 00:20:14 2002
+++ ./Makefile	Mon Oct  7 00:23:03 2002
@@ -16,7 +16,7 @@
 CXXFLAGS = -pipe -Wall -W -O2 -march=athlon -I/usr/local/include/libpng12 -D_REENTRANT -fPIC  -DQT_NO_DEBUG -DQT_THREAD_SUPPORT -DQT_PLUGIN
 LEXFLAGS = 
 YACCFLAGS= -d
-INCPATH  = -I$(QTDIR)/include -I$(QTDIR)/share/qt/mkspecs/linux-g++
+INCPATH  = -I$(QTDIR)/include/qt -I$(QTDIR)/share/qt/mkspecs/linux-g++
 LINK     = g++
 LFLAGS   = -shared -Wl,-soname,libqssl.so.1
 LIBS     = $(SUBLIBS)  -Wl,-rpath,$(QTDIR)/lib -L$(QTDIR)/lib -lssl -lqt-mt -lpthread
