diff -urN psi-old/src/common.cpp psi/src/common.cpp
--- psi-old/src/common.cpp	2004-02-15 20:24:02.615978632 +0100
+++ psi/src/common.cpp	2004-02-15 20:24:11.393644224 +0100
@@ -1296,7 +1296,8 @@
 
 QString getHomeDir()
 {
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
@@ -1387,7 +1388,8 @@
 
 QString getHomeDir()
 {
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
