diff -ru -X exclude psi-clean/src/contactview.cpp psi/src/contactview.cpp
--- psi-clean/src/contactview.cpp	Mon Jan 26 17:33:30 2004
+++ psi/src/contactview.cpp	Sun Feb 01 20:12:08 2004
@@ -1878,7 +1878,12 @@
 			item->contactProfile()->scActionDefault(item);
 	}
 	else {
-		if(option.useleft) {
+		const QPixmap * pix = item->pixmap(0);			
+		if(button == QListView::LeftButton && item->type() == ContactViewItem::Group &&
+		   pix && viewport()->mapFromGlobal(pos).x() <= pix->width() + treeStepSize()) {
+			setOpen(item, !item->isOpen());
+		}
+		else if(option.useleft) {
 			if(button == QListView::LeftButton) {
 				if(option.singleclick) {
 					qlv_contextPopup(i, pos, c);
