--- psi/src/common.cpp~	2004-03-10 21:03:53.000000000 +0100
+++ psi/src/common.cpp	2004-03-21 17:56:22.993517712 +0100
@@ -1160,7 +1160,8 @@
 
 		LinuxPLD, // Polish Linux distros
 		LinuxAurox,
-		LinuxArch
+		LinuxArch,
+		LinuxCustom
 	};
 
 	enum OsFlags {
@@ -1175,6 +1176,7 @@
 		QString file;
 		QString name;
 	} osInfo[] = {
+		{ LinuxCustom,		OsUseFile,	getHomeDir()+"/custom-os",	"",			},
 		{ LinuxMandrake,	OsUseFile,	"/etc/mandrake-release",	"Mandrake Linux"	},
 		{ LinuxDebian,		OsAppendFile,	"/etc/debian_version",		"Debian GNU/Linux"	},
 		{ LinuxGentoo,		OsUseFile,	"/etc/gentoo-release",		"Gentoo Linux"		},
