diff -Nur psi-0.9-orig/src/common.cpp psi-0.9/src/common.cpp
--- psi-0.9-orig/src/common.cpp	Tue Jun 17 05:18:39 2003
+++ psi-0.9/src/common.cpp	Thu Jun 19 12:23:50 2003
@@ -1227,6 +1227,7 @@
 		LinuxGentoo,
 		LinuxSlackware,
 		LinuxSuSE,
+		LinuxPLD,
 
 		LinuxASP, // russian Linux distros
 		LinuxALT
@@ -1252,6 +1253,7 @@
 
 		// untested
 		{ LinuxSuSE,		OsUseName,	"/etc/rc.config.d",		"SuSE Linux"		},
+		{ LinuxPLD,		OsUseName,	"/etc/pld-release",		"PLD Linux (Ra)"		},
 
 		{ LinuxNone,		OsUseName,	"",				""			}
 	};
