diff -urN psi-old/src/psi_profiles.cpp psi/src/psi_profiles.cpp
--- psi-old/src/psi_profiles.cpp	2004-02-16 01:28:59.895369360 +0100
+++ psi/src/psi_profiles.cpp	2004-02-16 01:29:11.992530312 +0100
@@ -434,7 +434,7 @@
 	prefs.onevent[eSend]    = g.pathBase + "/sound/send.wav";
 
 	// Added by Kiko 020621: sets up the default certificate store
-	prefs.trustCertStoreDir = g.pathBase + "/certs";
+	prefs.trustCertStoreDir = g.pathHome + "/certs";
 
 	prefs.sizeEventDlg = EventDlg::defaultSize();
 	prefs.sizeChatDlg = ChatDlg::defaultSize();
