--- psi-0.12/src/psiaccount.cpp.orig	2008-05-14 23:33:03.000000000 +0200
+++ psi-0.12/src/psiaccount.cpp	2008-08-01 17:56:28.000000000 +0200
@@ -2988,11 +2988,11 @@
 void PsiAccount::changeStatus(int x)
 {
 	if(x == STATUS_OFFLINE && !PsiOptions::instance()->getOption("options.status.ask-for-message-on-offline").toBool()) {
-		setStatus(Status(Status::Offline, "Logged out", 0));
+		setStatus(Status("","",0,true));
 	}
 	else {
 		if(x == STATUS_ONLINE && !PsiOptions::instance()->getOption("options.status.ask-for-message-on-online").toBool()) {
-			setStatus(Status());
+			setStatus(Status("","",0,true));
 		}
 		else if(x == STATUS_INVISIBLE){
 			Status s("","",0,true);
