diff -ru -X exclude iris-clean/xmpp-im/types.cpp iris/xmpp-im/types.cpp
--- iris-clean/xmpp-im/types.cpp	Wed Nov 26 18:24:42 2003
+++ iris/xmpp-im/types.cpp	Thu May 20 12:12:28 2004
@@ -941,7 +1026,7 @@
 
 bool RosterItem::addGroup(const QString &g)
 {
-	if(inGroup(g))
+	if(inGroup(g) || g.isEmpty())
 		return false;
 
 	v_groups += g;
@@ -990,8 +1075,10 @@
 		QDomElement i = n.toElement();
 		if(i.isNull())
 			continue;
-		if(i.tagName() == "group")
+		if(i.tagName() == "group"){
 			g += tagContent(i);
+			g.remove(QString(""));
+		}
 	}
 	QString a = item.attribute("ask");
 
