--- psi-0.11/src/psi_profiles.cpp.orig	2007-10-14 16:00:53.000000000 +0200
+++ psi-0.11/src/psi_profiles.cpp	2008-01-25 16:39:11.509545735 +0100
@@ -561,7 +561,7 @@
 	prefs.onevent[eFTComplete] = ApplicationInfo::resourcesDir() + "/sound/ft_complete.wav";
 
 	// Added by Kiko 020621: sets up the default certificate store
-	prefs.trustCertStoreDir = ApplicationInfo::resourcesDir() + "/certs";
+	prefs.trustCertStoreDir = ApplicationInfo::homeDir() + "/certs";
 
 	prefs.sizeEventDlg = EventDlg::defaultSize();
 	prefs.sizeChatDlg = ChatDlg::defaultSize();
