--- psi-0.9.4/psi/src/contactview.cpp	2005-03-27 16:29:31.000000000 +0200
+++ psi-0.9.4.plus/psi/src/contactview.cpp	2005-03-27 16:45:57.000000000 +0200
@@ -2723,6 +2723,25 @@
 		if(!ins.isEmpty()){
 			repl.setPattern("<!version>"); txt.replace(repl, QString("<br>") + ins);
 		}
+    } else {
+		// status description
+		ins = linkify(expandEntities(d->u->lastUnavailableStatus().status().stripWhiteSpace()));
+		if(option.useEmoticons)
+			ins = emoticonify(ins);
+		ins.replace("\n", "<br>");
+		if(option.templFieldTrim > 0 ){
+			// try not to break <tags>
+			int trim = ins.left(option.templFieldTrim).findRev("<");
+			if( trim != -1 )
+				trim = ins.find(">", trim);
+			if( trim == -1 || trim < option.templFieldTrim)
+				trim = option.templFieldTrim;
+			ins = ins.left(trim).stripWhiteSpace() + (((int)ins.length() > trim)?(" ..."):(""));
+		}
+		repl.setPattern("<status_msg>"); txt.replace(repl, ins);
+		if(!ins.isEmpty()){
+			repl.setPattern("<!status_msg>"); txt.replace(repl, QString("<br>") + ins);
+		}
 	}
 	// ... TODO
 
