--- psi-0.11/src/psicon.cpp.orig	2007-10-14 16:00:53.000000000 +0200
+++ psi-0.11/src/psicon.cpp	2008-01-26 00:05:49.668941951 +0100
@@ -918,8 +918,8 @@
 
 void PsiCon::setToggles(bool tog_offline, bool tog_away, bool tog_agents, bool tog_hidden, bool tog_self)
 {
-	if(d->contactList->enabledAccounts().count() > 1)
-		return;
+//	if(d->contactList->enabledAccounts().count() > 1)
+//		return;
 
 	d->mainwin->cvlist->setShowOffline(tog_offline);
 	d->mainwin->cvlist->setShowAway(tog_away);
