diff -aurN psi-0.9.1.orig/src/common.cpp psi-0.9.1/src/common.cpp
--- psi-0.9.1.orig/src/common.cpp	2004-01-02 02:23:05.000000000 +0100
+++ psi-0.9.1/src/common.cpp	2004-02-13 20:05:35.261388472 +0100
@@ -1283,7 +1283,8 @@
 
 QString getHomeDir()
 {
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
@@ -1374,7 +1375,8 @@
 
 QString getHomeDir()
 {
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
