--- psi-0.11/src/applicationinfo.cpp.orig	2007-10-14 16:00:51.000000000 +0200
+++ psi-0.11/src/applicationinfo.cpp	2008-01-25 16:32:03.390665615 +0100
@@ -131,7 +131,8 @@
 		return p;
 
 #if defined(Q_WS_X11)
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL && (strncmp(home_etc, "", 1) != 0) ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
@@ -160,7 +161,8 @@
 
 	return proghome.path();
 #elif defined(Q_WS_MAC)
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL && (strncmp(home_etc, "", 1) != 0) ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
