diff -urN psi-old/src/common.cpp psi/src/common.cpp
--- psi-old/src/common.cpp	2004-02-15 20:19:22.990488168 +0100
+++ psi/src/common.cpp	2004-02-15 20:21:23.576156360 +0100
@@ -1162,7 +1162,9 @@
 		LinuxALT,
 
 		LinuxPLD, // Polish Linux distros
-		LinuxAurox
+		LinuxAurox,
+
+                LinuxCustom // custom distro string,  ~/.psi/custom-os
 	};
 
 	enum OsFlags {
@@ -1177,6 +1179,7 @@
 		QString file;
 		QString name;
 	} osInfo[] = {
+		{ LinuxCustom,		OsUseFile,	getHomeDir()+"/custom-os",	"",			},
 		{ LinuxMandrake,	OsUseFile,	"/etc/mandrake-release",	"Mandrake Linux"	},
 		{ LinuxDebian,		OsAppendFile,	"/etc/debian_version",		"Debian GNU/Linux"	},
 		{ LinuxGentoo,		OsUseFile,	"/etc/gentoo-release",		"Gentoo Linux"		},
