diff -aurN psi-0.9.orig/src/common.h psi-0.9/src/common.h
--- psi-0.9.orig/src/common.h	2003-06-16 20:58:47.000000000 +0200
+++ psi-0.9/src/common.h	2003-06-18 17:38:31.000000000 +0200
@@ -61,14 +61,14 @@
 extern bool use_gpg, no_gpg_agent;
 
 // options stuff
-enum { cOnline, cOffline, cAway, cDND, cGroupFore, cGroupBack, cListBack };
+enum { cOnline, cOffline, cAway, cDND, cGroupFore, cGroupBack, cListBack, cAnimFront, cAnimBack };
 enum { fRoster, fMessage, fChat };
 enum { eMessage, eChat1, eChat2, eSystem, eOnline, eOffline, eSend };
 enum { dcClose, dcHour, dcDay, dcNever };
 
 struct Options
 {
-	QColor color[7];
+	QColor color[9];
 	QString font[3];
 	QString iconset;
 	int alertStyle;
diff -aurN psi-0.9.orig/src/contactview.cpp psi-0.9/src/contactview.cpp
--- psi-0.9.orig/src/contactview.cpp	2003-06-01 05:03:46.000000000 +0200
+++ psi-0.9/src/contactview.cpp	2003-06-18 17:38:41.000000000 +0200
@@ -1993,7 +1993,10 @@
 
 	setProfileState(STATUS_OFFLINE);
 	setText(0, profileName);
-	setOpen(true);
+
+	QString display = cp->d->pa->userAccount().contactViewData[profileName];
+    bool viewState = (display == QString::null) || (display.compare("open") == 0);
+    setOpen(viewState);
 }
 
 ContactViewItem::ContactViewItem(const QString &groupName, int groupType, ContactProfile *cp, ContactView *parent)
@@ -2008,7 +2011,9 @@
 
 	drawGroupIcon();
 	resetGroupName();
-	setOpen(true);
+	QString display = cp->d->pa->userAccount().contactViewData[groupName];
+    bool viewState = (display == QString::null) || (display.compare("open") == 0);
+    setOpen(viewState);
 	setDropEnabled(true);
 }
 
@@ -2027,7 +2032,9 @@
 
 	drawGroupIcon();
 	resetGroupName();
-	setOpen(true);
+	QString display = cp->d->pa->userAccount().contactViewData[groupName];
+    bool viewState = (display == QString::null) || (display.compare("open") == 0);
+    setOpen(viewState);
 	setDropEnabled(true);
 }
 
@@ -2163,8 +2170,8 @@
 			xcg.setColor(QColorGroup::Text, option.color[cOffline]);
 
 		if(d->animatingNick) {
-			xcg.setColor(QColorGroup::Text, d->animateNickColor ? Qt::red : Qt::black);
-			xcg.setColor(QColorGroup::HighlightedText, d->animateNickColor ? Qt::red : Qt::black);
+			xcg.setColor(QColorGroup::Text, d->animateNickColor ? option.color[cAnimFront] : option.color[cAnimBack]);
+			xcg.setColor(QColorGroup::HighlightedText, d->animateNickColor ? option.color[cAnimFront] : option.color[cAnimBack]);
 		}
 
 		QListViewItem::paintCell(p, xcg, column, width, alignment);
@@ -2185,16 +2192,18 @@
 	else if(d->type == Group || d->type == Profile) {
 		QColorGroup xcg = cg;
 		xcg.setColor(QColorGroup::Text, option.color[cGroupFore]);
-		xcg.setColor(QColorGroup::Base, option.color[cGroupBack]);
+//		xcg.setColor(QColorGroup::Base, option.color[cGroupBack]);
 
-		//QFont f = p->font();
-		//f.setPixelSize(8);
-		//p->setFont(f);
+		if(d->type == Group) {
+			QFont f = p->font();
+			f.setPointSize(f.pointSize() - 2);
+			p->setFont(f);
+		}
 
 		QListViewItem::paintCell(p, xcg, column, width, alignment);
 
-		p->setPen(QPen(option.color[cGroupFore]));
-		p->drawRect(0, 0, width, height());
+//		p->setPen(QPen(option.color[cGroupFore]));
+//		p->drawRect(0, 0, width, height());
 
 		QFontMetrics fm(p->font());
 		int x = fm.width(text(column)) + pixmap(column)->width() + 8;
@@ -2213,6 +2222,17 @@
 
 		int y = ((height() - fm.height()) / 2) + fm.ascent();
 		p->drawText(x, y, d->groupInfo);
+		
+		x += fm.width(d->groupInfo) + 8;
+		if(x < width - 8) {
+			int h = (height() / 2) - 1;
+			p->setPen(QPen(option.color[cGroupBack]));
+			p->drawLine(x, h, width - 8, h);
+			h++;
+			p->setPen(QPen(option.color[cGroupFore]));
+			p->drawLine(x, h, width - 8, h);
+		}
+		
 	}
 }
 
diff -aurN psi-0.9.orig/src/optionsdlg.cpp psi-0.9/src/optionsdlg.cpp
--- psi-0.9.orig/src/optionsdlg.cpp	2003-06-07 16:24:32.000000000 +0200
+++ psi-0.9/src/optionsdlg.cpp	2003-06-18 17:38:31.000000000 +0200
@@ -144,7 +144,9 @@
 	bg_color->insert(pb_cGroupFore, 4);
 	bg_color->insert(pb_cGroupBack, 5);
 	bg_color->insert(pb_cListBack, 6);
-	for(n = 0; n < 7; ++n) {
+	bg_color->insert(pb_cAnimFront, 7);
+	bg_color->insert(pb_cAnimBack, 8);
+	for(n = 0; n < 9; ++n) {
 		showColor((QPushButton *)bg_color->find(n), opt.color[n]);
 	}
 	connect(bg_color, SIGNAL(clicked(int)), this, SLOT(chooseColor(int)));
@@ -367,6 +369,10 @@
 		tr("Specifies the background color for a group name in the main window."));
 	QWhatsThis::add(pb_cListBack,
 		tr("Specifies the background color for the main window."));
+	QWhatsThis::add(pb_cAnimFront,
+		tr("Specifies the foreground animation color for nicks."));
+	QWhatsThis::add(pb_cAnimBack,
+		tr("Specifies the background animation color for nicks."));
 	QWhatsThis::add(le_font[0],
 		tr("Specifies the font style for the main window."));
 	QWhatsThis::add(le_font[1],
diff -aurN psi-0.9.orig/src/profiles.cpp psi-0.9/src/profiles.cpp
--- psi-0.9.orig/src/profiles.cpp	2003-05-26 02:40:13.000000000 +0200
+++ psi-0.9/src/profiles.cpp	2003-06-18 17:38:41.000000000 +0200
@@ -314,6 +314,15 @@
 		r.appendChild(tag);
 	}
 
+	QDomElement cd = doc.createElement("contactviewlistdata");
+	a.appendChild(cd);
+    for (QMap<QString, QString>::ConstIterator itr = contactViewData.begin(); itr != contactViewData.end(); itr++) {
+		QDomElement tag = doc.createElement("viewitemdata");
+        tag.setAttribute("group", itr.key());
+        tag.setAttribute("display", itr.data());
+        cd.appendChild(tag);
+    }
+
 	a.appendChild(textTag(doc, "proxyindex", QString::number(proxy_index)));
 	//a.appendChild(textTag(doc, "proxytype", QString::number(proxy_type)));
 	//a.appendChild(textTag(doc, "proxyhost", proxy_host));
@@ -396,6 +405,19 @@
 	else
 		opt_passphrase = false;
 
+	QDomElement cd = findSubTag(a, "contactviewlistdata", &found);
+	if(found) {
+		for(QDomNode n = cd.firstChild(); !n.isNull(); n = n.nextSibling()) {
+			QDomElement i = n.toElement();
+			if(i.isNull())
+				continue;
+
+			if(i.tagName() == "viewitemdata") {
+                contactViewData[i.attribute("group")] = i.attribute("display");
+			}
+		}
+	}
+
 	QDomElement r = findSubTag(a, "roster", &found);
 	if(found) {
 		for(QDomNode n = r.firstChild(); !n.isNull(); n = n.nextSibling()) {
@@ -513,6 +535,8 @@
 	prefs.color[cOffline]   = QColor("#000000");
 	prefs.color[cGroupFore] = QColor("#000000");
 	prefs.color[cGroupBack] = QColor("#FFFFFF");
+	prefs.color[cAnimFront] = QColor("#FF0000");
+	prefs.color[cAnimBack] 	= QColor("#000000");
 
 	prefs.font[fRoster] = QApplication::font().toString();
 	prefs.font[fMessage] = QApplication::font().toString();
@@ -755,6 +779,8 @@
 			tag.appendChild(textTag(doc, "offline", prefs.color[cOffline].name() ));
 			tag.appendChild(textTag(doc, "groupfore", prefs.color[cGroupFore].name() ));
 			tag.appendChild(textTag(doc, "groupback", prefs.color[cGroupBack].name() ));
+			tag.appendChild(textTag(doc, "animfront", prefs.color[cAnimFront].name() ));
+			tag.appendChild(textTag(doc, "animback", prefs.color[cAnimBack].name() ));
 		}
 
 		{
@@ -1090,6 +1116,8 @@
 				readColorEntry(tag, "offline", &prefs.color[cOffline]);
 				readColorEntry(tag, "groupfore", &prefs.color[cGroupFore]);
 				readColorEntry(tag, "groupback", &prefs.color[cGroupBack]);
+				readColorEntry(tag, "animfront", &prefs.color[cAnimFront]);
+				readColorEntry(tag, "animback", &prefs.color[cAnimBack]);
 			}
 
 			tag = findSubTag(p_lnf, "fonts", &found);
diff -aurN psi-0.9.orig/src/profiles.h psi-0.9/src/profiles.h
--- psi-0.9.orig/src/profiles.h	2003-05-26 02:40:13.000000000 +0200
+++ psi-0.9/src/profiles.h	2003-06-18 17:38:41.000000000 +0200
@@ -52,6 +52,7 @@
 
 	QString olr_string;
 	Jabber::Roster roster;
+    QMap<QString, QString> contactViewData;
 
 	bool opt_passphrase;
 	QString pgpPassphrase;
diff -aurN psi-0.9.orig/src/psiaccount.cpp psi-0.9/src/psiaccount.cpp
--- psi-0.9.orig/src/psiaccount.cpp	2003-06-17 05:46:02.000000000 +0200
+++ psi-0.9/src/psiaccount.cpp	2003-06-18 17:38:41.000000000 +0200
@@ -223,6 +223,8 @@
 
 	d->gcbank.setAutoDelete(true);
 
+    d->acc.contactViewData = acc.contactViewData;
+
 	// create Jabber::Client
 	d->client = new Client;
 	d->client->setOSName(getOSName());
@@ -281,6 +283,9 @@
 	connect(d->cp, SIGNAL(actionAssignKey(const Jid &)),SLOT(actionAssignKey(const Jid &)));
 	connect(d->cp, SIGNAL(actionUnassignKey(const Jid &)),SLOT(actionUnassignKey(const Jid &)));
 
+ 	connect(d->cp->contactView(), SIGNAL(expanded(QListViewItem*)), SLOT(listStateChage(QListViewItem*)));
+ 	connect(d->cp->contactView(), SIGNAL(collapsed(QListViewItem*)), SLOT(listStateChage(QListViewItem*)));
+
 	// restore cached roster
 	for(Roster::ConstIterator it = acc.roster.begin(); it != acc.roster.end(); ++it)
 		client_rosterItemUpdated(*it);
@@ -2893,6 +2898,22 @@
 		d->cp->removeSelf();
 }
 
+void PsiAccount::listStateChage(QListViewItem* litem)
+{
+    ContactViewItem* cvi = (ContactViewItem*)litem;
+    if (cvi->contactProfile()->name() != d->cp->name())
+        return;
+
+    QString isOpen = cvi->isOpen() ? "open" : "close";
+ 	int type = cvi->type();
+    if (type == ContactViewItem::Profile) {
+        d->acc.contactViewData[cvi->contactProfile()->name()] = isOpen;
+    }
+    else if (type == ContactViewItem::Group) {
+        d->acc.contactViewData[cvi->groupName()] = isOpen;
+    }
+}
+
 QString PsiAccount::resultToString(int result)
 {
 	QString s;
diff -aurN psi-0.9.orig/src/psiaccount.h psi-0.9/src/psiaccount.h
--- psi-0.9.orig/src/psiaccount.h	2003-06-16 22:18:27.000000000 +0200
+++ psi-0.9/src/psiaccount.h	2003-06-18 17:38:41.000000000 +0200
@@ -307,6 +307,8 @@
 
 	void optionsUpdate();
 
+    void listStateChage(QListViewItem*);
+
 private:
 	class Private;
 	Private *d;
diff -aurN psi-0.9.orig/src/ui_options.ui psi-0.9/src/ui_options.ui
--- psi-0.9.orig/src/ui_options.ui	2003-05-28 01:15:21.000000000 +0200
+++ psi-0.9/src/ui_options.ui	2003-06-18 17:38:31.000000000 +0200
@@ -1169,7 +1169,7 @@
                                         <number>11</number>
                                     </property>
                                     <property name="spacing">
-                                        <number>6</number>
+                                        <number>8</number>
                                     </property>
                                     <widget class="QPushButton" row="3" column="1">
                                         <property name="name">
@@ -1243,6 +1243,22 @@
                                             <string>List Background</string>
                                         </property>
                                     </widget>
+                                    <widget class="QLabel" row="7" column="0">
+                                        <property name="name">
+                                            <cstring>text3</cstring>
+                                        </property>
+                                        <property name="text">
+                                            <string>Nick Anim Foreground</string>
+                                        </property>
+                                    </widget>
+                                    <widget class="QLabel" row="8" column="0">
+                                        <property name="name">
+                                            <cstring>text4</cstring>
+                                        </property>
+                                        <property name="text">
+                                            <string>Nick Anim Background</string>
+                                        </property>
+                                    </widget>
                                     <widget class="QPushButton" row="0" column="1">
                                         <property name="name">
                                             <cstring>pb_cOnline</cstring>
@@ -1339,6 +1355,38 @@
                                             <string></string>
                                         </property>
                                     </widget>
+                                    <widget class="QPushButton" row="7" column="1">
+                                        <property name="name">
+                                            <cstring>pb_cAnimFront</cstring>
+                                        </property>
+                                        <property name="sizePolicy">
+                                            <sizepolicy>
+                                                <hsizetype>0</hsizetype>
+                                                <vsizetype>0</vsizetype>
+                                                <horstretch>0</horstretch>
+                                                <verstretch>0</verstretch>
+                                            </sizepolicy>
+                                        </property>
+                                        <property name="text">
+                                            <string></string>
+                                        </property>
+                                    </widget>
+                                    <widget class="QPushButton" row="8" column="1">
+                                        <property name="name">
+                                            <cstring>pb_cAnimBack</cstring>
+                                        </property>
+                                        <property name="sizePolicy">
+                                            <sizepolicy>
+                                                <hsizetype>0</hsizetype>
+                                                <vsizetype>0</vsizetype>
+                                                <horstretch>0</horstretch>
+                                                <verstretch>0</verstretch>
+                                            </sizepolicy>
+                                        </property>
+                                        <property name="text">
+                                            <string></string>
+                                        </property>
+                                    </widget>
                                 </grid>
                             </widget>
                             <widget class="QLayoutWidget">
