diff -urN psi-0.9.3.orig/iris/xmpp-im/client.cpp psi-0.9.3/iris/xmpp-im/client.cpp
--- psi-0.9.3.orig/iris/xmpp-im/client.cpp	2004-12-28 21:19:45.000000000 +0100
+++ psi-0.9.3/iris/xmpp-im/client.cpp	2005-01-13 23:47:01.314289544 +0100
@@ -792,10 +792,27 @@
 
 	// unavailable?  remove the resource
 	if(!s.isAvailable()) {
-		if(found) {
-			(*rit).setStatus(s);
-			debug(QString("Client: Removing resource from [%1]: name=[%2]\n").arg(i->jid().full()).arg(j.resource()));
-			resourceUnavailable(j, *rit);
+		Resource r;
+		if(!found) {
+			//if (!s.status().isEmpty()) {
+				r = Resource(j.resource(), s);
+				i->resourceList() += r;
+				debug(QString("Client: Adding resource to [%1]: name=[%2]\n").arg(i->jid().full()).arg(j.resource()));
+				resourceAvailable(j, r);
+		} 
+		else {
+				(*rit).setStatus(s);
+				r = *rit;
+				debug(QString("Client: Updating resource to [%1]: name=[%2]\n").arg(i->jid().full()).arg(j.resource()));
+				resourceAvailable(j, r);			
+		}
+		
+		rit = i->resourceList().find(j.resource());
+		found = (rit == i->resourceList().end()) ? false: true;
+		
+ 		if(found) {
+ 			(*rit).setStatus(s);
+ 			debug(QString("Client: Removing resource from [%1]: name=[%2]\n").arg(i->jid().full()).arg(j.resource())); 	resourceUnavailable(j, *rit);
 			i->resourceList().remove(rit);
 			i->setLastUnavailableStatus(s);
 		}
diff -urN psi-0.9.3.orig/src/chatdlg.cpp psi-0.9.3/src/chatdlg.cpp
--- psi-0.9.3.orig/src/chatdlg.cpp	2004-12-22 13:37:35.000000000 +0100
+++ psi-0.9.3/src/chatdlg.cpp	2005-01-13 23:47:01.341285440 +0100
@@ -474,8 +474,10 @@
 			u = ul.first();
 			if(rname.isEmpty()) {
 				// use priority
-				if(!u->isAvailable())
+				if(!u->isAvailable()){
 					status = STATUS_OFFLINE;
+					statusString = u->lastUnavailableStatus().status();
+				}
 				else {
 					const UserResource &r = *u->userResourceList().priority();
 					status = makeSTATUS(r.status());
diff -urN psi-0.9.3.orig/src/common.h psi-0.9.3/src/common.h
--- psi-0.9.3.orig/src/common.h	2004-12-22 13:37:35.000000000 +0100
+++ psi-0.9.3/src/common.h	2005-01-13 23:47:01.344284984 +0100
@@ -143,6 +143,7 @@
 	QStringList gcHighlights, gcNickColors;
 
 	bool clNewHeadings;
+	bool clStIndicator;
 
 	// passive popups
 	bool ppIsOn;
diff -urN psi-0.9.3.orig/src/contactview.cpp psi-0.9.3/src/contactview.cpp
--- psi-0.9.3.orig/src/contactview.cpp	2004-10-29 19:00:38.000000000 +0200
+++ psi-0.9.3/src/contactview.cpp	2005-01-13 23:47:01.351283920 +0100
@@ -25,6 +25,7 @@
 #include<qheader.h>
 #include<qtimer.h>
 #include<qpainter.h>
+#include<qbitmap.h>
 #include<qpopupmenu.h>
 #include<qmessagebox.h>
 #include<qinputdialog.h>
@@ -2766,12 +2767,13 @@
 
 void ContactViewItem::setIcon(const Icon *icon, bool alert)
 {
+	/* needed by status indicator :o|
 	if ( d->lastIcon == icon ) {
 		return; // cause less flicker. but still have to run calltree valgring skin on psi while online (mblsha).
 	}
 	else
 		d->lastIcon = (Icon *)icon;
-
+	*/
 	if ( d->icon ) {
 		disconnect(d->icon, 0, this, 0 );
 		d->icon->stop();
@@ -2793,12 +2795,42 @@
 		pix = d->icon->pixmap();
 	}
 
-	setPixmap(0, pix);
+	iconUpdated(pix);
 }
 
 void ContactViewItem::iconUpdated(const QPixmap &pix)
 {
-	setPixmap(0, pix);
+	QPixmap out(pix);
+	// add status indicator to contact icon
+	if(d->u && !out.isNull()){
+		bool ind = false;
+		if(!d->u->userResourceList().isEmpty() && option.clStIndicator) {
+			UserResourceList srl = d->u->userResourceList();
+
+			for(UserResourceList::ConstIterator rit = srl.begin(); rit != srl.end(); ++rit)
+				if(!(*rit).status().status().isEmpty()) { ind = true; break; }
+			
+		} 
+		else if(!d->u->lastUnavailableStatus().status().isEmpty() && option.clStIndicator)
+			ind = true;
+
+		if(ind) {
+			const QPixmap &ipix = is->indicator(d->u->jid()).pixmap();
+			// sum masks
+			if(out.hasAlpha() && ipix.hasAlpha()){
+				QBitmap mask(*out.mask());
+				const QBitmap* imask = ipix.mask();
+				bitBlt(&mask, (mask.width() - imask->width())/2, (mask.height() - imask->height())/2, 
+					imask, 0, 0, imask->width(), imask->height(), 
+					QPixmap::OrROP);
+				out.setMask(mask);
+			}
+			bitBlt(&out, (out.width() - ipix.width())/2, (out.height() - ipix.height())/2, 
+				&ipix, 0, 0, ipix.width(), ipix.height(), 
+				QPixmap::CopyROP);
+		}
+	}
+	setPixmap(0, out);
 }
 
 void ContactViewItem::animateNick()
diff -urN psi-0.9.3.orig/src/options/opt_appearance.cpp psi-0.9.3/src/options/opt_appearance.cpp
--- psi-0.9.3.orig/src/options/opt_appearance.cpp	2004-11-20 13:03:44.000000000 +0100
+++ psi-0.9.3/src/options/opt_appearance.cpp	2005-01-13 23:49:08.758915016 +0100
@@ -147,6 +147,8 @@
 		tr("Specifies the foreground animation color for nicks."));
 	QWhatsThis::add(d->pb_cAnimBack,
 		tr("Specifies the background animation color for nicks."));
+	QWhatsThis::add(d->ck_statusIndicator,
+		tr("Displays an indicator by the contact name if the contact entered a descriptive text status line."));
 
 	// Avatars
 	//QWhatsThis::add(d->ck_avatarsChatdlg,
@@ -160,8 +162,9 @@
 	if ( !w )
 		return;
 
-	//OptAppearanceUI *d = (OptAppearanceUI *)w;
+	OptAppearanceUI *d = (OptAppearanceUI *)w;
 	//opt->avatarsChatdlgEnabled = d->ck_avatarsChatdlg->isChecked(); // Avatars
+	opt->clStIndicator = d->ck_statusIndicator->isChecked();
 
 	int n;
 	for (n = 0; n < 4; ++n)
@@ -176,9 +179,10 @@
 	if ( !w )
 		return;
 
-	//OptAppearanceUI *d = (OptAppearanceUI *)w;
+	OptAppearanceUI *d = (OptAppearanceUI *)w;
 	//d->ck_avatarsChatdlg->setChecked( opt->avatarsChatdlgEnabled ); // Avatars
-
+	d->ck_statusIndicator->setChecked( opt->clStIndicator );
+	
 	int n;
 	for (n = 0; n < 4; ++n)
 		le_font[n]->setFont(opt->font[n]);
diff -urN psi-0.9.3.orig/src/options/opt_appearance-ui.ui psi-0.9.3/src/options/opt_appearance-ui.ui
--- psi-0.9.3.orig/src/options/opt_appearance-ui.ui	2004-11-20 13:03:44.000000000 +0100
+++ psi-0.9.3/src/options/opt_appearance-ui.ui	2005-01-13 23:47:01.356283160 +0100
@@ -308,6 +308,14 @@
                 </widget>
             </grid>
         </widget>
+        <widget class="QCheckBox">
+            <property name="name">
+                <cstring>ck_statusIndicator</cstring>
+            </property>
+            <property name="text">
+                <string>Show indicator for text statuses</string>
+            </property>
+        </widget>
         <widget class="QGroupBox">
             <property name="name">
                 <cstring>groupBox12</cstring>
diff -urN psi-0.9.3.orig/src/psiiconset.cpp psi-0.9.3/src/psiiconset.cpp
--- psi-0.9.3.orig/src/psiiconset.cpp	2004-06-14 12:34:43.000000000 +0200
+++ psi-0.9.3/src/psiiconset.cpp	2005-01-13 23:47:01.370281032 +0100
@@ -516,6 +516,19 @@
 	return Icon();
 }
 
+Icon *PsiIconset::indicatorPtr(const XMPP::Jid &jid)
+{
+	return d->jid2icon(jid, "psi/statInd");
+}
+
+Icon PsiIconset::indicator(const XMPP::Jid &jid)
+{
+	Icon *icon = indicatorPtr(jid);
+	if ( icon )
+		return *icon;
+	return Icon();
+}
+
 Icon *PsiIconset::statusPtr(const XMPP::Jid &jid, int s)
 {
 	return d->jid2icon(jid, status2name(s));
diff -urN psi-0.9.3.orig/src/psiiconset.h psi-0.9.3/src/psiiconset.h
--- psi-0.9.3.orig/src/psiiconset.h	2003-11-20 04:32:47.000000000 +0100
+++ psi-0.9.3/src/psiiconset.h	2005-01-13 23:47:01.400276472 +0100
@@ -61,9 +61,11 @@
 	// JID-enabled status functions
 	Icon *statusPtr(const XMPP::Jid &, int);
 	Icon *statusPtr(const XMPP::Jid &, const XMPP::Status &);
+	Icon *indicatorPtr(const XMPP::Jid &);
 
 	Icon status(const XMPP::Jid &, int);
 	Icon status(const XMPP::Jid &, const XMPP::Status &);
+	Icon indicator(const XMPP::Jid &);
 
 	// functions to get status icon by transport name
 	Icon *transportStatusPtr(QString name, int);
diff -urN psi-0.9.3.orig/src/psi_profiles.cpp psi-0.9.3/src/psi_profiles.cpp
--- psi-0.9.3.orig/src/psi_profiles.cpp	2004-12-08 15:42:39.000000000 +0100
+++ psi-0.9.3/src/psi_profiles.cpp	2005-01-13 23:47:01.406275560 +0100
@@ -441,6 +441,8 @@
 		prefs.font[fPopup] = font.toString();
 	}
 
+	prefs.clStIndicator = TRUE;
+
 	// calculate the small font size
 	const int minimumFontSize = 7;
 	prefs.smallFontSize = qApp->font().pointSize();
@@ -877,6 +879,7 @@
 		p.appendChild(p_lnf);
 
 		p_lnf.appendChild(textTag(doc, "newHeadings", prefs.clNewHeadings));
+		p_lnf.appendChild(textTag(doc, "statusIndicator", prefs.clStIndicator));
 
 		{
 			QDomElement tag = doc.createElement("colors");
@@ -1383,6 +1386,7 @@
 			bool found;
 
 			readBoolEntry(p_lnf, "newHeadings", &prefs.clNewHeadings);
+			readBoolEntry(p_lnf, "statusIndicator", &prefs.clStIndicator);
 
 			QDomElement tag = findSubTag(p_lnf, "colors", &found);
 			if(found) {
