diff -aurN psi-0.9.1-test1.orig/psi.desktop psi-0.9.1-test1/psi.desktop
--- psi-0.9.1-test1.orig/psi.desktop	2003-03-13 01:27:46.000000000 +0100
+++ psi-0.9.1-test1/psi.desktop	2003-12-10 20:18:30.612412888 +0100
@@ -1,10 +1,12 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Psi
-GenericName=Jabber Client
-Exec=psi -caption "%c" %i %m
-Icon=psi
-Terminal=0
+Comment=Send and receive instant messages using the Jabber system
+Comment[pl]=Klient Jabbera
+Exec=psi
+Icon=psi.png
+Terminal=false
 Type=Application
-X-KDE-StartupNotify=true
+Categories=Qt;Network;InstantMessaging;
+Encoding=UTF-8
 
+# vi: encoding=utf-8
