diff -ru -X exclude psi-clean/libpsi/psiwidgets/iconwidget.cpp psi/libpsi/psiwidgets/iconwidget.cpp
--- psi-clean/libpsi/psiwidgets/iconwidget.cpp	Thu Dec 18 14:10:44 2003
+++ psi/libpsi/psiwidgets/iconwidget.cpp	Thu Dec 18 14:27:52 2003
@@ -587,11 +587,6 @@
 
 void IconButton::setIcon(const Icon *i, bool activate)
 {
-#ifndef Q_WS_X11
-	if ( !text().isEmpty() )
-		return;
-#endif
-
 	forceSetIcon(i, activate);
 	d->forced = false;
 }
