diff -urN psi-0.12.orig/options/default.xml psi-0.12/options/default.xml
--- psi-0.12.orig/options/default.xml	2008-05-13 03:57:35.000000000 +0200
+++ psi-0.12/options/default.xml	2008-08-09 22:01:13.000000000 +0200
@@ -214,6 +214,12 @@
 			<last-used-save-path type="QString"></last-used-save-path>
 			<look>
 				<colors>
+					<chat>
+						<incoming-messages-color type="QColor" >#0000ff</incoming-messages-color>
+						<outgoing-messages-color type="QColor" >#ff0000</outgoing-messages-color>
+						<spooled-messages-color type="QColor" >#008000</spooled-messages-color>
+						<system-messages-color type="QColor" >#00a000</system-messages-color>
+					</chat>
 					<contactlist>
 						<background type="QColor">#ffffff</background>
 						<grouping>
@@ -232,6 +238,12 @@
 						</status>
 						<status-messages type="QColor">#808080</status-messages>
 					</contactlist>
+					<groupchat>
+						<alert-messages-color type="QColor" >#ff0000</alert-messages-color>
+						<default-nick-color type="QColor" >#000000</default-nick-color>
+						<spooled-messages-color type="QColor" >#008000</spooled-messages-color>
+						<system-messages-color type="QColor" >#00a000</system-messages-color>
+					</groupchat>
 					<muc>
 						<nick-colors type="QStringList" >
 							<item>Blue</item>
diff -urN psi-0.12.orig/src/groupchatdlg.cpp psi-0.12/src/groupchatdlg.cpp
--- psi-0.12.orig/src/groupchatdlg.cpp	2008-03-19 21:59:10.000000000 +0100
+++ psi-0.12/src/groupchatdlg.cpp	2008-08-09 18:34:42.000000000 +0200
@@ -1207,7 +1207,7 @@
 	bool doInsert = t.date() != lastMsgTime_.date();
 	lastMsgTime_ = t;
 	if (doInsert) {
-		QString color = "#00A000";
+		QString color = PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.system-messages-color").toString();
 		ui_.log->appendText(QString("<font color=\"%1\">*** %2</font>").arg(color).arg(t.date().toString(Qt::ISODate)));
 	}
 }
@@ -1226,7 +1226,7 @@
 
 	updateLastMsgTime(time);
 	QString timestr = ui_.log->formatTimeStamp(time);
-	ui_.log->appendText(QString("<font color=\"#00A000\">[%1]").arg(timestr) + QString(" *** %1</font>").arg(Qt::escape(str)));
+	ui_.log->appendText(QString("<font color=\"" + PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.system-messages-color").toString() +  "\">[%1]").arg(timestr) + QString(" *** %1</font>").arg(Qt::escape(str)));
 
 	if(alert)
 		doAlert();
@@ -1249,7 +1249,7 @@
 	QStringList nickColors = PsiOptions::instance()->getOption("options.ui.look.colors.muc.nick-colors").toStringList();
 	
 	if(!PsiOptions::instance()->getOption("options.ui.muc.use-nick-coloring").toBool() || nickColors.empty()) {
-		return "#000000";
+		return PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.default-nick-color").toString();
 	}
 	else if(sender == -1 || nickColors.size() == 1) {
 		return nickColors[nickColors.size()-1];
@@ -1280,12 +1280,12 @@
 	nickcolor = getNickColor(who);
 	textcolor = ui_.log->palette().active().text().name();
 	if(alert) {
-		textcolor = "#FF0000";
+		textcolor = PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.alert-messages-color").toString();
 		alerttagso = "<b>";
 		alerttagsc = "</b>";
 	}
 	if(m.spooled())
-		nickcolor = "#008000"; //color = "#008000";
+		nickcolor = PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.spooled-messages-color").toString();
 
 	QString timestr = ui_.log->formatTimeStamp(m.timeStamp());
 
diff -urN psi-0.12.orig/src/options/opt_appearance.cpp psi-0.12/src/options/opt_appearance.cpp
--- psi-0.12.orig/src/options/opt_appearance.cpp	2008-01-12 03:48:36.000000000 +0100
+++ psi-0.12/src/options/opt_appearance.cpp	2008-08-09 21:09:33.000000000 +0200
@@ -17,6 +17,7 @@
 #include <qlineedit.h>
 
 #include "ui_opt_appearance.h"
+#include "ui_opt_appearance_fonts.h"
 #include "ui_opt_appearance_misc.h"
 #include "psioptions.h"
 
@@ -27,6 +28,12 @@
 	OptAppearanceUI() : QWidget() { setupUi(this); }
 };
 
+class OptAppearanceFontsUI : public QWidget, public Ui::OptAppearanceFonts
+{
+public:
+	OptAppearanceFontsUI() : QWidget() { setupUi(this); }
+};
+
 class OptAppearanceMiscUI : public QWidget, public Ui::OptAppearanceMisc
 {
 public:
@@ -72,6 +79,7 @@
 OptionsTabAppearance::OptionsTabAppearance(QObject *parent) : MetaOptionsTab(parent, "appearance", "", tr("Appearance"), tr("Psi's Appearance"), "psi/appearance")
 {    
 	addTab( new OptionsTabAppearanceGeneral(this) );
+	addTab( new OptionsTabAppearanceFonts(this) );
 	addTab( new OptionsTabIconsetEmoticons(this) );
 	addTab( new OptionsTabIconsetRoster(this) );
 	addTab( new OptionsTabIconsetSystem(this) );
@@ -135,21 +143,18 @@
 }
 
 //----------------------------------------------------------------------------
-// OptionsTabAppearanceGeneral: Fonts & Colours
+// OptionsTabAppearanceGeneral: Colours
 //----------------------------------------------------------------------------
 
 OptionsTabAppearanceGeneral::OptionsTabAppearanceGeneral(QObject *parent)
-: OptionsTab(parent, "appearance_general", "", tr("Fonts && Colors"), tr("Fonts && Color Settings"))
+: OptionsTab(parent, "appearance_general", "", tr("Colors"), tr("Color Settings"))
 {
 	w = 0;
-	bg_font = 0;
 	bg_color = 0;
 }
 
 OptionsTabAppearanceGeneral::~OptionsTabAppearanceGeneral()
 {
-	if ( bg_font )
-		delete bg_font;
 	if ( bg_color )
 		delete bg_color;
 }
@@ -175,33 +180,6 @@
 	w = new OptAppearanceUI();
 	OptAppearanceUI *d = (OptAppearanceUI *)w;
 
-	le_font[0] = d->le_fRoster;
-	le_font[1] = d->le_fMessage;
-	le_font[2] = d->le_fChat;
-	le_font[3] = d->le_fPopup;
-
-	bg_font = new QButtonGroup;
-	bg_font->insert(d->pb_fRoster);
-	bg_font->insert(d->pb_fMessage);
-	bg_font->insert(d->pb_fChat);
-	bg_font->insert(d->pb_fPopup);
-	connect(bg_font, SIGNAL(buttonClicked(QAbstractButton*)), SLOT(chooseFont(QAbstractButton*)));
-
-	QWhatsThis::add(le_font[0],
-		tr("Specifies the font style for the main window."));
-	QWhatsThis::add(le_font[1],
-		tr("Specifies the font style for message windows."));
-	QWhatsThis::add(le_font[2],
-		tr("Specifies the font style for chat windows."));
-	QWhatsThis::add(le_font[3],
-		tr("Specifies the font style for popup windows."));
-	QWhatsThis::add(d->pb_fRoster,
-		tr("Selects a font for the roster window using the font selection dialog."));
-	QWhatsThis::add(d->pb_fMessage,
-		tr("Selects a font for message windows using the font selection dialog."));
-	QWhatsThis::add(d->pb_fChat,
-		tr("Selects a font for chat windows using the font selection dialog."));
-
 	bg_color = new QButtonGroup;
 	bg_color->insert(d->pb_cOnline);
 	bg_color->insert(d->pb_cOffline);
@@ -214,6 +192,15 @@
 	bg_color->insert(d->pb_cListBack);
 	bg_color->insert(d->pb_cAnimFront);
 	bg_color->insert(d->pb_cAnimBack);
+	bg_color->insert(d->pb_cChatSys);
+	bg_color->insert(d->pb_cChatSpool);
+	bg_color->insert(d->pb_cChatTo);
+	bg_color->insert(d->pb_cChatFrom);
+	bg_color->insert(d->pb_cGChatAlert);
+	bg_color->insert(d->pb_cGChatSys);
+	bg_color->insert(d->pb_cGChatSpool);
+	bg_color->insert(d->pb_cGChatNick);
+
 	bg_color->insert(d->pb_cStatus);
 	connect(bg_color, SIGNAL(buttonClicked(QAbstractButton*)), SLOT(chooseColor(QAbstractButton*)));
 
@@ -238,6 +225,22 @@
 		tr("Specifies the foreground animation color for nicks."));
 	QWhatsThis::add(d->pb_cAnimBack,
 		tr("Specifies the background animation color for nicks."));
+	QWhatsThis::add(d->pb_cChatSys,
+		tr("Specifies the color for system chat messages."));
+	QWhatsThis::add(d->pb_cChatSpool,
+		tr("Specifies the color for spooled chat messages."));
+	QWhatsThis::add(d->pb_cChatTo,
+		tr("Specifies the nick color for sent chat messages."));
+	QWhatsThis::add(d->pb_cChatFrom,
+		tr("Specifies the nick color for recieved chat messages."));
+	QWhatsThis::add(d->pb_cGChatAlert,
+		tr("Specifies the color for alert groupchat messages."));
+	QWhatsThis::add(d->pb_cGChatSys,
+		tr("Specifies the color for system groupchat messages."));
+	QWhatsThis::add(d->pb_cGChatSpool,
+		tr("Specifies the color for spooled groupchat messages."));
+	QWhatsThis::add(d->pb_cGChatNick,
+		tr("Specifies the default nick color for groupchat messages."));
 
 	// Avatars
 	//QWhatsThis::add(d->ck_avatarsChatdlg,
@@ -265,11 +268,6 @@
 	OptAppearanceUI *d = (OptAppearanceUI *)w;
 	//LEGOPTS.avatarsChatdlgEnabled = d->ck_avatarsChatdlg->isChecked(); // Avatars
 
-	PsiOptions::instance()->setOption("options.ui.look.font.contactlist", d->le_fRoster->fontName());
-	PsiOptions::instance()->setOption("options.ui.look.font.message", d->le_fMessage->fontName());
-	PsiOptions::instance()->setOption("options.ui.look.font.chat", d->le_fChat->fontName());
-	PsiOptions::instance()->setOption("options.ui.look.font.passive-popup", d->le_fPopup->fontName());
-
 	PsiOptions::instance()->setOption("options.ui.look.colors.contactlist.status.online", getColor(d->pb_cOnline));
 	PsiOptions::instance()->setOption("options.ui.look.colors.contactlist.status.offline", getColor(d->pb_cOffline));
 	PsiOptions::instance()->setOption("options.ui.look.colors.contactlist.status.away", getColor(d->pb_cAway));
@@ -281,6 +279,14 @@
 	PsiOptions::instance()->setOption("options.ui.look.colors.contactlist.background", getColor(d->pb_cListBack));
 	PsiOptions::instance()->setOption("options.ui.look.contactlist.status-change-animation.color1", getColor(d->pb_cAnimFront));
 	PsiOptions::instance()->setOption("options.ui.look.contactlist.status-change-animation.color2", getColor(d->pb_cAnimBack));
+	PsiOptions::instance()->setOption("options.ui.look.colors.chat.system-messages-color", getColor(d->pb_cChatSys));
+	PsiOptions::instance()->setOption("options.ui.look.colors.chat.spooled-messages-color", getColor(d->pb_cChatSpool));
+	PsiOptions::instance()->setOption("options.ui.look.colors.chat.outgoing-messages-color", getColor(d->pb_cChatTo));
+	PsiOptions::instance()->setOption("options.ui.look.colors.chat.incoming-messages-color", getColor(d->pb_cChatFrom));
+	PsiOptions::instance()->setOption("options.ui.look.colors.groupchat.alert-messages-color", getColor(d->pb_cGChatAlert));
+	PsiOptions::instance()->setOption("options.ui.look.colors.groupchat.system-messages-color", getColor(d->pb_cGChatSys));
+	PsiOptions::instance()->setOption("options.ui.look.colors.groupchat.spooled-messages-color", getColor(d->pb_cGChatSpool));
+	PsiOptions::instance()->setOption("options.ui.look.colors.groupchat.default-nick-color", getColor(d->pb_cGChatNick));
 	PsiOptions::instance()->setOption("options.ui.look.colors.contactlist.status-messages", getColor(d->pb_cStatus));
 }
 
@@ -298,11 +304,6 @@
 	OptAppearanceUI *d = (OptAppearanceUI *)w;
 	//d->ck_avatarsChatdlg->setChecked( LEGOPTS.avatarsChatdlgEnabled ); // Avatars
 
-	d->le_fRoster->setFont(PsiOptions::instance()->getOption("options.ui.look.font.contactlist").toString());
-	d->le_fMessage->setFont(PsiOptions::instance()->getOption("options.ui.look.font.message").toString());
-	d->le_fChat->setFont(PsiOptions::instance()->getOption("options.ui.look.font.chat").toString());
-	d->le_fPopup->setFont(PsiOptions::instance()->getOption("options.ui.look.font.passive-popup").toString());
-
 	restoreColor(d->pb_cOnline, PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.status.online").value<QColor>());
 	restoreColor(d->pb_cOffline, PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.status.offline").value<QColor>());
 	restoreColor(d->pb_cAway, PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.status.away").value<QColor>());
@@ -314,6 +315,14 @@
 	restoreColor(d->pb_cListBack, PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.background").value<QColor>());
 	restoreColor(d->pb_cAnimFront, PsiOptions::instance()->getOption("options.ui.look.contactlist.status-change-animation.color1").value<QColor>());
 	restoreColor(d->pb_cAnimBack, PsiOptions::instance()->getOption("options.ui.look.contactlist.status-change-animation.color2").value<QColor>());
+	restoreColor(d->pb_cChatSys, PsiOptions::instance()->getOption("options.ui.look.colors.chat.system-messages-color").value<QColor>());
+	restoreColor(d->pb_cChatSpool, PsiOptions::instance()->getOption("options.ui.look.colors.chat.spooled-messages-color").value<QColor>());
+	restoreColor(d->pb_cChatTo, PsiOptions::instance()->getOption("options.ui.look.colors.chat.outgoing-messages-color").value<QColor>());
+	restoreColor(d->pb_cChatFrom, PsiOptions::instance()->getOption("options.ui.look.colors.chat.incoming-messages-color").value<QColor>());
+	restoreColor(d->pb_cGChatAlert, PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.alert-messages-color").value<QColor>());
+	restoreColor(d->pb_cGChatSys, PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.system-messages-color").value<QColor>());
+	restoreColor(d->pb_cGChatSpool, PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.spooled-messages-color").value<QColor>());
+	restoreColor(d->pb_cGChatNick, PsiOptions::instance()->getOption("options.ui.look.colors.groupchat.default-nick-color").value<QColor>());
 	restoreColor(d->pb_cStatus, PsiOptions::instance()->getOption("options.ui.look.colors.contactlist.status-messages").value<QColor>());
 }
 
@@ -331,26 +331,6 @@
 	parentWidget = parentDialog;
 }
 
-void OptionsTabAppearanceGeneral::chooseFont(QAbstractButton* button)
-{
-	bool ok;
-	QFont font;
-	int x = (bg_font->buttons()).indexOf(button);
-	font.fromString( le_font[x]->fontName() );
-
-	// ensure we don't use the new native font dialog on mac with Qt 4.5,
-	//   since it was broken last we checked (qt task #252000)
-#if QT_VERSION >= 0x040500
-	QString fnt = QFontDialog::getFont(&ok, font, parentWidget, QString(), QFontDialog::DontUseNativeDialog).toString();
-#else
-	QString fnt = QFontDialog::getFont(&ok, font, parentWidget).toString();
-#endif
-	le_font[x]->setFont(fnt);
-
-	if(ok)
-		emit dataChanged();
-}
-
 void OptionsTabAppearanceGeneral::chooseColor(QAbstractButton* button)
 {
 	QColor c;
@@ -352,3 +347,106 @@
 		emit dataChanged();
 	}
 }
+
+//----------------------------------------------------------------------------
+// OptionsTabAppearanceFonts: Fonts
+//----------------------------------------------------------------------------
+
+OptionsTabAppearanceFonts::OptionsTabAppearanceFonts(QObject *parent)
+: OptionsTab(parent, "appearance_general", "", tr("Fonts"), tr("Fonts"))
+{
+	w = 0;
+	bg_font = 0;
+}
+
+OptionsTabAppearanceFonts::~OptionsTabAppearanceFonts()
+{
+	if ( bg_font )
+		delete bg_font;
+}
+
+QWidget *OptionsTabAppearanceFonts::widget()
+{
+	if ( w )
+		return 0;
+
+	w = new OptAppearanceFontsUI();
+	OptAppearanceFontsUI *d = (OptAppearanceFontsUI *)w;
+
+	le_font[0] = d->le_fRoster;
+	le_font[1] = d->le_fMessage;
+	le_font[2] = d->le_fChat;
+	le_font[3] = d->le_fPopup;
+
+	bg_font = new QButtonGroup;
+	bg_font->insert(d->pb_fRoster);
+	bg_font->insert(d->pb_fMessage);
+	bg_font->insert(d->pb_fChat);
+	bg_font->insert(d->pb_fPopup);
+	connect(bg_font, SIGNAL(buttonClicked(QAbstractButton*)), SLOT(chooseFont(QAbstractButton*)));
+
+	QWhatsThis::add(le_font[0],
+		tr("Specifies the font style for the main window."));
+	QWhatsThis::add(le_font[1],
+		tr("Specifies the font style for message windows."));
+	QWhatsThis::add(le_font[2],
+		tr("Specifies the font style for chat windows."));
+	QWhatsThis::add(le_font[3],
+		tr("Specifies the font style for popup windows."));
+	QWhatsThis::add(d->pb_fRoster,
+		tr("Selects a font for the roster window using the font selection dialog."));
+	QWhatsThis::add(d->pb_fMessage,
+		tr("Selects a font for message windows using the font selection dialog."));
+	QWhatsThis::add(d->pb_fChat,
+		tr("Selects a font for chat windows using the font selection dialog."));
+	
+	return w;
+}
+
+
+void OptionsTabAppearanceFonts::applyOptions()
+{
+	if ( !w )
+		return;
+
+	OptAppearanceFontsUI *d = (OptAppearanceFontsUI *)w;
+	//LEGOPTS.avatarsChatdlgEnabled = d->ck_avatarsChatdlg->isChecked(); // Avatars
+
+	PsiOptions::instance()->setOption("options.ui.look.font.contactlist", d->le_fRoster->fontName());
+	PsiOptions::instance()->setOption("options.ui.look.font.message", d->le_fMessage->fontName());
+	PsiOptions::instance()->setOption("options.ui.look.font.chat", d->le_fChat->fontName());
+	PsiOptions::instance()->setOption("options.ui.look.font.passive-popup", d->le_fPopup->fontName());
+}
+
+void OptionsTabAppearanceFonts::restoreOptions()
+{
+	if ( !w )
+		return;
+
+	OptAppearanceFontsUI *d = (OptAppearanceFontsUI *)w;
+	//d->ck_avatarsChatdlg->setChecked( LEGOPTS.avatarsChatdlgEnabled ); // Avatars
+
+	d->le_fRoster->setFont(PsiOptions::instance()->getOption("options.ui.look.font.contactlist").toString());
+	d->le_fMessage->setFont(PsiOptions::instance()->getOption("options.ui.look.font.message").toString());
+	d->le_fChat->setFont(PsiOptions::instance()->getOption("options.ui.look.font.chat").toString());
+	d->le_fPopup->setFont(PsiOptions::instance()->getOption("options.ui.look.font.passive-popup").toString());
+}
+
+void OptionsTabAppearanceFonts::setData(PsiCon *, QWidget *parentDialog)
+{
+	parentWidget = parentDialog;
+}
+
+void OptionsTabAppearanceFonts::chooseFont(QAbstractButton* button)
+{
+	bool ok;
+	QFont font;
+	int x = (bg_font->buttons()).indexOf(button);
+	font.fromString( le_font[x]->fontName() );
+
+	QString fnt = QFontDialog::getFont(&ok, font, parentWidget).toString();
+	le_font[x]->setFont(fnt);
+
+	if(ok)
+		emit dataChanged();
+}
diff -urN psi-0.12.orig/src/options/opt_appearance_fonts.ui psi-0.12/src/options/opt_appearance_fonts.ui
--- psi-0.12.orig/src/options/opt_appearance_fonts.ui	1970-01-01 01:00:00.000000000 +0100
+++ psi-0.12/src/options/opt_appearance_fonts.ui	2008-08-09 18:34:42.000000000 +0200
@@ -0,0 +1,142 @@
+<ui version="4.0" >
+ <author></author>
+ <comment></comment>
+ <exportmacro></exportmacro>
+ <class>OptAppearanceFonts</class>
+ <widget class="QWidget" name="OptAppearanceFonts" >
+  <property name="geometry" >
+   <rect>
+    <x>0</x>
+    <y>0</y>
+    <width>402</width>
+    <height>184</height>
+   </rect>
+  </property>
+  <property name="windowTitle" >
+   <string>OptAppearanceFontsUI</string>
+  </property>
+  <layout class="QVBoxLayout" >
+   <property name="margin" >
+    <number>9</number>
+   </property>
+   <property name="spacing" >
+    <number>6</number>
+   </property>
+   <item>
+    <widget class="QGroupBox" name="groupBox12" >
+     <property name="title" >
+      <string>Fonts</string>
+     </property>
+     <layout class="QGridLayout" >
+      <property name="margin" >
+       <number>8</number>
+      </property>
+      <property name="spacing" >
+       <number>6</number>
+      </property>
+      <item row="0" column="0" >
+       <widget class="QLabel" name="tl_fRoster" >
+        <property name="text" >
+         <string>Roster</string>
+        </property>
+       </widget>
+      </item>
+      <item row="1" column="0" >
+       <widget class="QLabel" name="tl_fMessage" >
+        <property name="text" >
+         <string>Message</string>
+        </property>
+       </widget>
+      </item>
+      <item row="2" column="0" >
+       <widget class="QLabel" name="tl_fChat" >
+        <property name="enabled" >
+         <bool>true</bool>
+        </property>
+        <property name="text" >
+         <string>Chat</string>
+        </property>
+       </widget>
+      </item>
+      <item row="3" column="0" >
+       <widget class="QLabel" name="tl_fPopup" >
+        <property name="text" >
+         <string>Popup</string>
+        </property>
+       </widget>
+      </item>
+      <item row="0" column="1" colspan="3" >
+       <widget class="FontLabel" name="le_fRoster" />
+      </item>
+      <item row="1" column="1" colspan="3" >
+       <widget class="FontLabel" name="le_fMessage" />
+      </item>
+      <item row="2" column="1" colspan="3" >
+       <widget class="FontLabel" name="le_fChat" />
+      </item>
+      <item row="3" column="1" colspan="3" >
+       <widget class="FontLabel" name="le_fPopup" />
+      </item>
+      <item row="0" column="4" >
+       <widget class="QPushButton" name="pb_fRoster" >
+        <property name="text" >
+         <string>Choose</string>
+        </property>
+       </widget>
+      </item>
+      <item row="1" column="4" >
+       <widget class="QPushButton" name="pb_fMessage" >
+        <property name="text" >
+         <string>Choose</string>
+        </property>
+       </widget>
+      </item>
+      <item row="2" column="4" >
+       <widget class="QPushButton" name="pb_fChat" >
+        <property name="enabled" >
+         <bool>true</bool>
+        </property>
+        <property name="text" >
+         <string>Choose</string>
+        </property>
+       </widget>
+      </item>
+      <item row="3" column="4" >
+       <widget class="QPushButton" name="pb_fPopup" >
+        <property name="enabled" >
+         <bool>true</bool>
+        </property>
+        <property name="text" >
+         <string>Choose</string>
+        </property>
+       </widget>
+      </item>
+     </layout>
+    </widget>
+   </item>
+  </layout>
+ </widget>
+ <layoutdefault spacing="6" margin="11" />
+ <pixmapfunction></pixmapfunction>
+ <customwidgets>
+  <customwidget>
+   <class>FontLabel</class>
+   <extends></extends>
+   <header>opt_appearance.h</header>
+   <container>0</container>
+   <pixmap></pixmap>
+  </customwidget>
+ </customwidgets>
+ <tabstops>
+  <tabstop>le_fRoster</tabstop>
+  <tabstop>pb_fRoster</tabstop>
+  <tabstop>le_fMessage</tabstop>
+  <tabstop>pb_fMessage</tabstop>
+  <tabstop>le_fChat</tabstop>
+  <tabstop>pb_fChat</tabstop>
+  <tabstop>le_fPopup</tabstop>
+  <tabstop>pb_fPopup</tabstop>
+ </tabstops>
+ <resources/>
+ <connections/>
+</ui>
diff -urN psi-0.12.orig/src/options/opt_appearance.h psi-0.12/src/options/opt_appearance.h
--- psi-0.12.orig/src/options/opt_appearance.h	2008-01-12 01:45:25.000000000 +0100
+++ psi-0.12/src/options/opt_appearance.h	2008-08-09 18:34:42.000000000 +0200
@@ -65,11 +65,29 @@
 private slots:
 	void setData(PsiCon *, QWidget *);
 	void chooseColor(QAbstractButton* button);
-	void chooseFont(QAbstractButton* button);
 
 private:
 	QWidget *w, *parentWidget;
 	QButtonGroup *bg_color;
+};
+
+class OptionsTabAppearanceFonts : public OptionsTab
+{
+	Q_OBJECT
+public:
+	OptionsTabAppearanceFonts(QObject *parent);
+	~OptionsTabAppearanceFonts();
+
+	QWidget *widget();
+	void applyOptions();
+	void restoreOptions();
+
+private slots:
+	void setData(PsiCon *, QWidget *);
+	void chooseFont(QAbstractButton* button);
+
+private:
+	QWidget *w, *parentWidget;
 	FontLabel *le_font[4];
 	QButtonGroup *bg_font;
 };
diff -urN psi-0.12.orig/src/options/opt_appearance.ui psi-0.12/src/options/opt_appearance.ui
--- psi-0.12.orig/src/options/opt_appearance.ui	2007-05-07 20:00:07.000000000 +0200
+++ psi-0.12/src/options/opt_appearance.ui	2008-08-09 18:34:42.000000000 +0200
@@ -8,8 +8,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>402</width>
-    <height>333</height>
+    <width>474</width>
+    <height>310</height>
    </rect>
   </property>
   <property name="windowTitle" >
@@ -34,63 +34,99 @@
       <property name="spacing" >
        <number>6</number>
       </property>
-      <item row="0" column="3" >
+      <item row="0" column="0" >
+       <widget class="QLabel" name="tl_cProfileFore" >
+        <property name="text" >
+         <string>Account heading foreground:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="1" column="0" >
+       <widget class="QLabel" name="tl_cProfileBack" >
+        <property name="text" >
+         <string>Account heading background:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="2" column="0" >
+       <widget class="QLabel" name="tl_cGroupFore" >
+        <property name="text" >
+         <string>Group heading foreground:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="3" column="0" >
+       <widget class="QLabel" name="tl_cGroupBack" >
+        <property name="text" >
+         <string>Group heading background:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="4" column="0" >
+       <widget class="QLabel" name="tl_cAnimFront" >
+        <property name="text" >
+         <string>Nick animation foreground</string>
+        </property>
+       </widget>
+      </item>
+      <item row="5" column="0" >
+       <widget class="QLabel" name="tl_cAnimBack" >
+        <property name="text" >
+         <string>Nick animation background:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="6" column="0" >
        <widget class="QLabel" name="tl_cOnline" >
         <property name="text" >
          <string>Online contacts:</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="3" >
+      <item row="7" column="0" >
        <widget class="QLabel" name="TextLabel1_2" >
         <property name="text" >
          <string>Away contacts:</string>
         </property>
        </widget>
       </item>
-      <item row="2" column="3" >
+      <item row="8" column="0" >
        <widget class="QLabel" name="tl_cDND" >
         <property name="text" >
          <string>DND contacts:</string>
         </property>
        </widget>
       </item>
-      <item row="3" column="3" >
+      <item row="9" column="0" >
        <widget class="QLabel" name="tl_cOffline" >
         <property name="text" >
          <string>Offline contacts:</string>
         </property>
        </widget>
       </item>
-      <item row="3" column="2" >
-       <spacer>
-        <property name="orientation" >
-         <enum>Qt::Horizontal</enum>
+      <item row="0" column="1" >
+       <widget class="QToolButton" name="pb_cProfileFore" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
         </property>
-        <property name="sizeHint" >
+        <property name="maximumSize" >
          <size>
-          <width>111</width>
+          <width>20</width>
           <height>20</height>
          </size>
         </property>
-       </spacer>
-      </item>
-      <item row="4" column="3" >
-       <widget class="QLabel" name="tl_cListBack" >
-        <property name="text" >
-         <string>Contact list background:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="5" column="3" >
-       <widget class="QLabel" name="tl_cStatus" >
         <property name="text" >
-         <string>Status messages:</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="0" column="4" >
-       <widget class="QToolButton" name="pb_cOnline" >
+      <item row="1" column="1" >
+       <widget class="QToolButton" name="pb_cProfileBack" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -110,8 +146,8 @@
         </property>
        </widget>
       </item>
-      <item row="1" column="4" >
-       <widget class="QToolButton" name="pb_cAway" >
+      <item row="2" column="1" >
+       <widget class="QToolButton" name="pb_cGroupFore" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -131,22 +167,29 @@
         </property>
        </widget>
       </item>
-      <item row="1" column="0" >
-       <widget class="QLabel" name="tl_cProfileBack" >
-        <property name="text" >
-         <string>Account heading background:</string>
+      <item row="3" column="1" >
+       <widget class="QToolButton" name="pb_cGroupBack" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="maximumSize" >
+         <size>
+          <width>20</width>
+          <height>20</height>
+         </size>
         </property>
-       </widget>
-      </item>
-      <item row="3" column="0" >
-       <widget class="QLabel" name="tl_cGroupBack" >
         <property name="text" >
-         <string>Group heading background:</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="5" column="1" >
-       <widget class="QToolButton" name="pb_cAnimBack" >
+      <item row="4" column="1" >
+       <widget class="QToolButton" name="pb_cAnimFront" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -166,8 +209,8 @@
         </property>
        </widget>
       </item>
-      <item row="1" column="1" >
-       <widget class="QToolButton" name="pb_cProfileBack" >
+      <item row="5" column="1" >
+       <widget class="QToolButton" name="pb_cAnimBack" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -187,8 +230,8 @@
         </property>
        </widget>
       </item>
-      <item row="3" column="1" >
-       <widget class="QToolButton" name="pb_cGroupBack" >
+      <item row="6" column="1" >
+       <widget class="QToolButton" name="pb_cOnline" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -208,8 +251,8 @@
         </property>
        </widget>
       </item>
-      <item row="0" column="1" >
-       <widget class="QToolButton" name="pb_cProfileFore" >
+      <item row="7" column="1" >
+       <widget class="QToolButton" name="pb_cAway" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -229,8 +272,8 @@
         </property>
        </widget>
       </item>
-      <item row="2" column="1" >
-       <widget class="QToolButton" name="pb_cGroupFore" >
+      <item row="8" column="1" >
+       <widget class="QToolButton" name="pb_cDND" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -250,8 +293,8 @@
         </property>
        </widget>
       </item>
-      <item row="4" column="1" >
-       <widget class="QToolButton" name="pb_cAnimFront" >
+      <item row="9" column="1" >
+       <widget class="QToolButton" name="pb_cOffline" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -271,36 +314,91 @@
         </property>
        </widget>
       </item>
-      <item row="4" column="0" >
-       <widget class="QLabel" name="tl_cAnimFront" >
+      <item row="3" column="2" >
+       <spacer>
+        <property name="orientation" >
+         <enum>Qt::Horizontal</enum>
+        </property>
+        <property name="sizeHint" >
+         <size>
+          <width>111</width>
+          <height>20</height>
+         </size>
+        </property>
+       </spacer>
+      </item>
+      <item row="0" column="3" >
+       <widget class="QLabel" name="tl_cListBack" >
         <property name="text" >
-         <string>Nick animation foreground:</string>
+         <string>Contact list background:</string>
         </property>
        </widget>
       </item>
-      <item row="5" column="0" >
-       <widget class="QLabel" name="tl_cAnimBack" >
+      <item row="1" column="3" >
+       <widget class="QLabel" name="tl_cStatus" >
         <property name="text" >
-         <string>Nick animation background:</string>
+         <string>Status messages:</string>
         </property>
        </widget>
       </item>
-      <item row="0" column="0" >
-       <widget class="QLabel" name="tl_cProfileFore" >
+      <item row="2" column="3" >
+       <widget class="QLabel" name="tl_cChatSys" >
         <property name="text" >
-         <string>Account heading foreground:</string>
+         <string>System chat messages:</string>
         </property>
        </widget>
       </item>
-      <item row="2" column="0" >
-       <widget class="QLabel" name="tl_cGroupFore" >
+      <item row="3" column="3" >
+       <widget class="QLabel" name="tl_cChatSpool" >
         <property name="text" >
-         <string>Group heading foreground:</string>
+         <string>Spooled chat messages:</string>
         </property>
        </widget>
       </item>
-      <item row="2" column="4" >
-       <widget class="QToolButton" name="pb_cDND" >
+      <item row="4" column="3" >
+       <widget class="QLabel" name="tl_cChatTo" >
+        <property name="text" >
+         <string>Outgoing chat message nick:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="5" column="3" >
+       <widget class="QLabel" name="tl_cChatFrom" >
+        <property name="text" >
+         <string>Incoming chat message nick:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="6" column="3" >
+       <widget class="QLabel" name="tl_cGChatAler" >
+        <property name="text" >
+         <string>Groupchat alert messages:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="7" column="3" >
+       <widget class="QLabel" name="tl_cGChatSys" >
+        <property name="text" >
+         <string>Groupchat system messages:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="8" column="3" >
+       <widget class="QLabel" name="tl_cGChatSpool" >
+        <property name="text" >
+         <string>Groupchat spooled messages:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="9" column="3" >
+       <widget class="QLabel" name="tl_cGChatNick" >
+        <property name="text" >
+         <string>Groupchat non-colored nicks:</string>
+        </property>
+       </widget>
+      </item>
+      <item row="0" column="4" >
+       <widget class="QToolButton" name="pb_cListBack" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -320,8 +418,8 @@
         </property>
        </widget>
       </item>
-      <item row="3" column="4" >
-       <widget class="QToolButton" name="pb_cOffline" >
+      <item row="1" column="4" >
+       <widget class="QToolButton" name="pb_cStatus" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -341,8 +439,8 @@
         </property>
        </widget>
       </item>
-      <item row="4" column="4" >
-       <widget class="QToolButton" name="pb_cListBack" >
+      <item row="2" column="4" >
+       <widget class="QToolButton" name="pb_cChatSys" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -362,8 +460,8 @@
         </property>
        </widget>
       </item>
-      <item row="5" column="4" >
-       <widget class="QToolButton" name="pb_cStatus" >
+      <item row="3" column="4" >
+       <widget class="QToolButton" name="pb_cChatSpool" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>0</hsizetype>
@@ -383,98 +481,132 @@
         </property>
        </widget>
       </item>
-     </layout>
-    </widget>
-   </item>
-   <item>
-    <widget class="QGroupBox" name="groupBox12" >
-     <property name="title" >
-      <string>Fonts</string>
-     </property>
-     <layout class="QGridLayout" >
-      <property name="margin" >
-       <number>8</number>
-      </property>
-      <property name="spacing" >
-       <number>6</number>
-      </property>
-      <item row="2" column="1" colspan="3" >
-       <widget class="FontLabel" name="le_fChat" />
-      </item>
-      <item row="2" column="0" >
-       <widget class="QLabel" name="tl_fChat" >
-        <property name="enabled" >
-         <bool>true</bool>
+      <item row="4" column="4" >
+       <widget class="QToolButton" name="pb_cChatTo" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
         </property>
-        <property name="text" >
-         <string>Chat:</string>
+        <property name="maximumSize" >
+         <size>
+          <width>20</width>
+          <height>20</height>
+         </size>
         </property>
-       </widget>
-      </item>
-      <item row="1" column="0" >
-       <widget class="QLabel" name="tl_fMessage" >
         <property name="text" >
-         <string>Message:</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="1" column="4" >
-       <widget class="QPushButton" name="pb_fMessage" >
-        <property name="text" >
-         <string>Choose...</string>
+      <item row="5" column="4" >
+       <widget class="QToolButton" name="pb_cChatFrom" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
         </property>
-       </widget>
-      </item>
-      <item row="2" column="4" >
-       <widget class="QPushButton" name="pb_fChat" >
-        <property name="enabled" >
-         <bool>true</bool>
+        <property name="maximumSize" >
+         <size>
+          <width>20</width>
+          <height>20</height>
+         </size>
         </property>
         <property name="text" >
-         <string>Choose...</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="3" column="4" >
-       <widget class="QPushButton" name="pb_fPopup" >
-        <property name="enabled" >
-         <bool>true</bool>
+      <item row="6" column="4" >
+       <widget class="QToolButton" name="pb_cGChatAlert" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="maximumSize" >
+         <size>
+          <width>20</width>
+          <height>20</height>
+         </size>
         </property>
         <property name="text" >
-         <string>Choose...</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="0" column="4" >
-       <widget class="QPushButton" name="pb_fRoster" >
+      <item row="7" column="4" >
+       <widget class="QToolButton" name="pb_cGChatSys" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="maximumSize" >
+         <size>
+          <width>20</width>
+          <height>20</height>
+         </size>
+        </property>
         <property name="text" >
-         <string>Choose...</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="0" column="0" >
-       <widget class="QLabel" name="tl_fRoster" >
+      <item row="8" column="4" >
+       <widget class="QToolButton" name="pb_cGChatSpool" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="maximumSize" >
+         <size>
+          <width>20</width>
+          <height>20</height>
+         </size>
+        </property>
         <property name="text" >
-         <string>Roster:</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="3" column="0" >
-       <widget class="QLabel" name="tl_fPopup" >
+      <item row="9" column="4" >
+       <widget class="QToolButton" name="pb_cGChatNick" >
+        <property name="sizePolicy" >
+         <sizepolicy>
+          <hsizetype>0</hsizetype>
+          <vsizetype>0</vsizetype>
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="maximumSize" >
+         <size>
+          <width>20</width>
+          <height>20</height>
+         </size>
+        </property>
         <property name="text" >
-         <string>Popup:</string>
+         <string/>
         </property>
        </widget>
       </item>
-      <item row="3" column="1" colspan="3" >
-       <widget class="FontLabel" name="le_fPopup" />
-      </item>
-      <item row="1" column="1" colspan="3" >
-       <widget class="FontLabel" name="le_fMessage" />
-      </item>
-      <item row="0" column="1" colspan="3" >
-       <widget class="FontLabel" name="le_fRoster" />
-      </item>
      </layout>
     </widget>
    </item>
@@ -504,14 +636,6 @@
   <tabstop>pb_cOffline</tabstop>
   <tabstop>pb_cListBack</tabstop>
   <tabstop>pb_cStatus</tabstop>
-  <tabstop>le_fRoster</tabstop>
-  <tabstop>pb_fRoster</tabstop>
-  <tabstop>le_fMessage</tabstop>
-  <tabstop>pb_fMessage</tabstop>
-  <tabstop>le_fChat</tabstop>
-  <tabstop>pb_fChat</tabstop>
-  <tabstop>le_fPopup</tabstop>
-  <tabstop>pb_fPopup</tabstop>
  </tabstops>
  <resources/>
  <connections/>
diff -urN psi-0.12.orig/src/options/options.pri psi-0.12/src/options/options.pri
--- psi-0.12.orig/src/options/options.pri	2008-02-11 23:24:22.000000000 +0100
+++ psi-0.12/src/options/options.pri	2008-08-09 18:34:42.000000000 +0200
@@ -46,6 +46,7 @@
 	$$PWD/opt_events.ui \
 	$$PWD/opt_status.ui \
 	$$PWD/opt_appearance.ui \
+	$$PWD/opt_appearance_fonts.ui \
 	$$PWD/opt_appearance_misc.ui \
 	$$PWD/opt_sound.ui \
 	$$PWD/opt_advanced.ui \
diff -urN psi-0.12.orig/src/psichatdlg.cpp psi-0.12/src/psichatdlg.cpp
--- psi-0.12.orig/src/psichatdlg.cpp	2008-02-02 18:59:35.000000000 +0100
+++ psi-0.12/src/psichatdlg.cpp	2008-08-09 18:34:42.000000000 +0200
@@ -420,7 +420,7 @@
 	QDateTime t = QDateTime::currentDateTime();
 	updateLastMsgTime(t);
 	QString timestr = chatView()->formatTimeStamp(t);
-	QString color = "#00A000";
+	QString color = PsiOptions::instance()->getOption("options.ui.look.colors.chat.system-messages-color").toString();
 
 	chatView()->appendText(QString("<font color=\"%1\">[%2]").arg(color, timestr) + QString(" *** %1</font>").arg(str));
 }
@@ -428,12 +428,12 @@
 QString PsiChatDlg::colorString(bool local, ChatDlg::SpooledType spooled) const
 {
 	if (spooled == ChatDlg::Spooled_OfflineStorage)
-		return "#008000";
+		return PsiOptions::instance()->getOption("options.ui.look.colors.chat.spooled-messages-color").toString();
 
 	if (local)
-		return "#FF0000";
+		return PsiOptions::instance()->getOption("options.ui.look.colors.chat.outgoing-messages-color").toString();
 
-	return "#0000FF";
+	return PsiOptions::instance()->getOption("options.ui.look.colors.chat.incoming-messages-color").toString();
 }
 
 ChatView* PsiChatDlg::chatView() const
@@ -459,7 +459,7 @@
 	bool doInsert = t.date() != lastMsgTime_.date();
 	lastMsgTime_ = t;
 	if (doInsert) {
-		QString color = "#00A000";
+		QString color = PsiOptions::instance()->getOption("options.ui.look.colors.chat.system-messages-color").toString();
 		chatView()->appendText(QString("<font color=\"%1\">*** %2</font>").arg(color).arg(t.date().toString(Qt::ISODate)));
 	}
 }
diff -urN psi-0.12.orig/src/psi_profiles.cpp psi-0.12/src/psi_profiles.cpp
--- psi-0.12.orig/src/psi_profiles.cpp	2008-04-24 20:06:44.000000000 +0200
+++ psi-0.12/src/psi_profiles.cpp	2008-08-09 21:55:03.000000000 +0200
@@ -1231,6 +1239,14 @@
 				migrateColorEntry(tag, "profileback", "options.ui.look.colors.contactlist.profile.header-background");
 				migrateColorEntry(tag, "animfront", "options.ui.look.contactlist.status-change-animation.color1");
 				migrateColorEntry(tag, "animback", "options.ui.look.contactlist.status-change-animation.color2");
+				migrateColorEntry(tag, "chatsys", "options.ui.look.colors.chat.system-messages-color");
+				migrateColorEntry(tag, "chatspool", "options.ui.look.colors.chat.spooled-messages-color");
+				migrateColorEntry(tag, "chatto", "options.ui.look.colors.chat.outgoing-messages-color");
+				migrateColorEntry(tag, "chatfrom", "options.ui.look.colors.chat.incoming-messages-color");
+				migrateColorEntry(tag, "gchatalert", "options.ui.look.colors.groupchat.alert-messages-color");
+				migrateColorEntry(tag, "gchatsys", "options.ui.look.colors.groupchat.system-messages-color");
+				migrateColorEntry(tag, "gchatspool", "options.ui.look.colors.groupchat.spooled-messages-color");
+				migrateColorEntry(tag, "gchatnick", "options.ui.look.colors.groupchat.default-nick-color");
 			}
 
 			tag = findSubTag(p_lnf, "fonts", &found);
