--- psi-0.8.7/src/kmp/stream.cpp.orig	Wed Oct 30 11:09:35 2002
+++ psi-0.8.7/src/kmp/stream.cpp	Sat Dec  7 00:42:19 2002
@@ -31,7 +31,7 @@
 
 #include"ndns.h"
 #include"xmlfilter.h"
-#include"../qssl/qssl.h"
+#include<qssl.h>
 
 using namespace Jabber;
 
--- psi-0.8.7/src/psi.pro.orig	Wed Oct 30 11:09:34 2002
+++ psi-0.8.7/src/psi.pro	Mon Dec  9 12:47:05 2002
@@ -153,8 +153,8 @@
 
 # System-specific build settings
 unix:!mac {
-	INCLUDEPATH  += /usr/include /usr/X11R6/include
-	LIBS         += -lz -lXss
+	INCLUDEPATH  += /usr/X11R6/include /usr/X11R6/include/qt
+	LIBS         += -lz -lXss -lqssl -L$(QTDIR)/lib/qt/plugins-mt/network -Wl,-rpath,$(QTDIR)/lib/qt/plugins-mt/network
 	SOURCES      += trayicon_x11.cpp
 
 	QMAKE_STRIP = "/bin/true"
