diff -urN psi-old/psi.desktop psi/psi.desktop
--- psi-old/psi.desktop	2004-02-15 20:23:12.345620888 +0100
+++ psi/psi.desktop	2004-02-15 20:23:29.850959672 +0100
@@ -1,10 +1,12 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Psi
-GenericName=Jabber Client
-Exec=psi -caption "%c" %i %m
-Icon=psi
-Terminal=0
+Comment=Send and receive instant messages using the Jabber system
+Comment[pl]=Klient Jabbera
+Exec=psi
+Icon=psi.png
+Terminal=false
 Type=Application
-X-KDE-StartupNotify=true
+Categories=Qt;Network;InstantMessaging;
+Encoding=UTF-8
 
+# vi: encoding=utf-8
