--- psi-0.11/src/systeminfo.cpp.orig	2007-10-14 16:00:54.000000000 +0200
+++ psi-0.11/src/systeminfo.cpp	2008-01-25 17:54:13.547934021 +0100
@@ -15,6 +15,7 @@
 #include <windows.h>
 #endif
 
+#include "applicationinfo.h"
 #include "systeminfo.h"
 
 SystemInfo::SystemInfo() : QObject(QCoreApplication::instance())
@@ -52,6 +53,7 @@
 	// get description about os
 	enum LinuxName {
 		LinuxNone = 0,
+		LinuxCustom,
 
 		LinuxMandrake,
 		LinuxDebian,
@@ -83,6 +85,8 @@
 		QString file;
 		QString name;
 	} osInfo[] = {
+		{ LinuxCustom,		OsUseFile,	ApplicationInfo::homeDir() + "/custom-os",	""	},
+
 		{ LinuxMandrake,	OsUseFile,	"/etc/mandrake-release",	"Mandrake Linux"	},
 		{ LinuxDebian,		OsAppendFile,	"/etc/debian_version",		"Debian GNU/Linux"	},
 		{ LinuxGentoo,		OsUseFile,	"/etc/gentoo-release",		"Gentoo Linux"		},
