--- psi-0.11/src/systeminfo.cpp.orig	2007-10-14 16:00:54.000000000 +0200
+++ psi-0.11/src/systeminfo.cpp	2008-01-25 16:36:04.883197197 +0100
@@ -52,6 +52,7 @@
 	// get description about os
 	enum LinuxName {
 		LinuxNone = 0,
+		LinuxCustom,
 
 		LinuxMandrake,
 		LinuxDebian,
@@ -100,6 +101,7 @@
 		// many distros use the /etc/redhat-release for compatibility, so RedHat will be the last :)
 		{ LinuxRedHat,		OsUseFile,	"/etc/redhat-release",		"RedHat Linux"		},
 
+		{ LinuxCustom,		OsUseFile,	ApplicationInfo::homeDir() + "/custom-os",	""	},
 		{ LinuxNone,		OsUseName,	"",				""			}
 	};
 
