diff -urN psi-0.9.3.orig/iris/xmpp-im/types.cpp psi-0.9.3/iris/xmpp-im/types.cpp
--- psi-0.9.3.orig/iris/xmpp-im/types.cpp	2004-10-14 21:14:40.000000000 +0200
+++ psi-0.9.3/iris/xmpp-im/types.cpp	2005-01-14 23:04:31.138456536 +0100
@@ -1026,7 +1026,7 @@
 
 bool RosterItem::addGroup(const QString &g)
 {
-	if(inGroup(g))
+	if(inGroup(g) || g.isEmpty())
 		return false;
 
 	v_groups += g;
@@ -1075,8 +1075,10 @@
 		QDomElement i = n.toElement();
 		if(i.isNull())
 			continue;
-		if(i.tagName() == "group")
+		if(i.tagName() == "group"){
 			g += tagContent(i);
+			g.remove(QString(""));
+		}
 	}
 	QString a = item.attribute("ask");
 
