diff -Nru psi-0.9.orig/src/common.cpp psi-0.9/src/common.cpp
--- psi-0.9.orig/src/common.cpp	2003-11-19 12:27:37.000000000 +0100
+++ psi-0.9/src/common.cpp	2003-11-19 12:39:22.000000000 +0100
@@ -1356,7 +1356,8 @@
 
 QString getHomeDir()
 {
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
@@ -1447,7 +1448,8 @@
 
 QString getHomeDir()
 {
-	QDir proghome(QDir::homeDirPath() + "/.psi");
+	char *home_etc = getenv("HOME_ETC");
+	QDir proghome((home_etc != NULL ? QString(home_etc) : QDir::homeDirPath()) + "/.psi");
 	if(!proghome.exists()) {
 		QDir home = QDir::home();
 		home.mkdir(".psi");
