diff -ru -X exclude psi-clean/src/contactview.cpp psi/src/contactview.cpp
--- psi-clean/src/contactview.cpp	Fri Mar 05 21:36:02 2004
+++ psi/src/contactview.cpp	Mon Mar 15 22:49:14 2004
@@ -1736,6 +1736,22 @@
 	delete d;
 }
 
+void ContactView::viewportPaintEvent( QPaintEvent* pe )
+{
+    QWidget* vp = viewport();
+
+	QSize sz = vp->size();
+    QPixmap pm(sz);
+    QPainter p(&pm, vp);
+	QRect r = vp->rect();
+	int ex = r.x() + contentsX();
+	int ey = r.y() + contentsY();
+	int ew = r.width();
+	int eh = r.height();
+	drawContentsOffset(&p, contentsX(), contentsY(), ex, ey, ew, eh);
+	bitBlt(vp, 0, 0, &pm);
+}
+
 QTimer *ContactView::animTimer() const
 {
 	return d->animTimer;
diff -ru -X exclude psi-clean/src/contactview.h psi/src/contactview.h
--- psi-clean/src/contactview.h	Tue Feb 24 20:46:24 2004
+++ psi/src/contactview.h	Mon Mar 15 22:35:34 2004
@@ -189,6 +189,7 @@
 	void maybeTip(const QPoint &);
 	void keyPressEvent(QKeyEvent *);
 	QDragObject *dragObject();
+	virtual void viewportPaintEvent(QPaintEvent *);
 
 signals:
 	void showOffline(bool);
