diff -ur psi-0.11.orig/src/widgets/iconwidget.cpp psi-0.11/src/widgets/iconwidget.cpp
--- psi-0.11.orig/src/widgets/iconwidget.cpp	2007-10-14 16:00:55.000000000 +0200
+++ psi-0.11/src/widgets/iconwidget.cpp	2007-11-26 20:32:28.000000000 +0100
@@ -642,11 +642,6 @@
 
 void IconButton::setPsiIcon(const PsiIcon *i, bool activate)
 {
-#ifndef Q_WS_X11
-	if ( !text().isEmpty() )
-		return;
-#endif
-
 	forceSetPsiIcon(i, activate);
 	d->forced = false;
 }
