--- configure.orig	2007-10-14 16:00:49.000000000 +0200
+++ configure	2008-01-25 15:31:19.691535871 +0100
@@ -701,7 +701,7 @@
 			if(QFile::exists(path))
 			{
 				QString certPathString = 
-					"QCA_SYSTEMSTORE_PATH=\\\\\\\\\\\\\"" + path + "\\\\\\\\\\\\\"";
+					"QCA_SYSTEMSTORE_PATH=" + path + "";
 				conf->addDefine(certPathString);
 				return true;
 			}
@@ -746,7 +746,7 @@
 
 		// Qt<4.2 workaround
 		QString certPathString = 
-			"QCA_SYSTEMSTORE_PATH=\\\\\\\\\\\\\"" + path + "\\\\\\\\\\\\\"";
+			"QCA_SYSTEMSTORE_PATH=" + path + "";
 		conf->addDefine(certPathString);
 
 		return true;
@@ -1142,8 +1142,8 @@
 		QFile file("src/config.h");
 		if ( file.open(QIODevice::WriteOnly | QIODevice::Text) ) {
 			QTextStream stream( &file );
-			stream << "#define PSI_LIBDIR \"" << conf->getenv("LIBDIR") << "/psi\"" << endl;
-			stream << "#define PSI_DATADIR \"" << conf->getenv("DATADIR") << "/psi\"" << endl;
+			stream << "#define PSI_LIBDIR \\"" << conf->getenv("LIBDIR") << "/psi\\"" << endl;
+			stream << "#define PSI_DATADIR \\"" << conf->getenv("DATADIR") << "/psi\\"" << endl;
 		}
 
 		conf->addDefine("HAVE_CONFIG");

