diff -urN psi-old/src/common.cpp psi/src/common.cpp
--- psi-old/src/common.cpp	2004-03-22 11:48:01.079698856 +0100
+++ psi/src/common.cpp	2004-03-22 11:49:21.273507544 +0100
@@ -1145,6 +1145,7 @@
 	// get description about os
 	enum LinuxName {
 		LinuxNone = 0,
+                LinuxCustom,
 
 		LinuxMandrake,
 		LinuxDebian,
@@ -1175,6 +1176,7 @@
 		QString file;
 		QString name;
 	} osInfo[] = {
+		{ LinuxCustom,		OsUseFile,	getHomeDir()+"/custom-os",	"",			},
 		{ LinuxMandrake,	OsUseFile,	"/etc/mandrake-release",	"Mandrake Linux"	},
 		{ LinuxDebian,		OsAppendFile,	"/etc/debian_version",		"Debian GNU/Linux"	},
 		{ LinuxGentoo,		OsUseFile,	"/etc/gentoo-release",		"Gentoo Linux"		},
