diff -ru -X exclude psi-clean/src/psicon.cpp psi/src/psicon.cpp
--- psi-clean/src/psicon.cpp	Fri Oct 24 05:38:42 2003
+++ psi/src/psicon.cpp	Sun Oct 26 14:42:44 2003
@@ -739,10 +739,7 @@
 		}
 		else {
 			QString str;
-			if(d->lastStatusString.isEmpty())
-				str = tr("I am away from my computer.  Please leave a message.");
-			else
-				str = d->lastStatusString; // default to old away message
+			str = d->lastStatusString; // default to old away message
 
 			StatusSetDlg *w = new StatusSetDlg(this, makeStatus(x, str));
 			connect(w, SIGNAL(set(const Status &)), SLOT(setStatusFromDialog(const Status &)));
