diff -aurN psi-0.9.1.orig/src/common.cpp psi-0.9.1/src/common.cpp
--- psi-0.9.1.orig/src/common.cpp	2004-01-02 02:23:05.000000000 +0100
+++ psi-0.9.1/src/common.cpp	2004-02-13 20:09:32.550315072 +0100
@@ -1155,7 +1155,8 @@
 		LinuxASP, // Russian Linux distros
 		LinuxALT,
 
-		LinuxPLD // Polish Linux distro
+		LinuxPLD, // Polish Linux distro
+		LinuxCustom // custom distro string, edit ~/.psi/custom-os
 	};
 
 	enum OsFlags {
@@ -1170,6 +1171,7 @@
 		QString file;
 		QString name;
 	} osInfo[] = {
+		{ LinuxCustom,		OsUseFile,	getHomeDir()+"/custom-os",	"",			},
 		{ LinuxMandrake,	OsUseFile,	"/etc/mandrake-release",	"Mandrake Linux"	},
 		{ LinuxDebian,		OsAppendFile,	"/etc/debian_version",		"Debian GNU/Linux"	},
 		{ LinuxRedHat,		OsUseFile,	"/etc/redhat-release",		"RedHat Linux"		},
