diff -ru psi-0.8.6/src/psi.pro psi-0.8.6-AlchemyX/src/psi.pro
--- psi-0.8.6/src/psi.pro	Mon Jul  8 20:13:11 2002
+++ psi-0.8.6-AlchemyX/src/psi.pro	Sat Nov 16 01:59:19 2002
@@ -140,7 +140,7 @@
 
 # System-specific build settings
 unix:!mac {
-	INCLUDEPATH  += /usr/include /usr/X11R6/include
+	INCLUDEPATH  += /usr/X11R6/include /usr/X11R6/include/qt
 	LIBS         += -lz -lXss
 	SOURCES      += trayicon_x11.cpp
 }
--- psi-0.8.7/src/kmp/stream.cpp.orig	Wed Oct 30 11:09:35 2002
+++ psi-0.8.7/src/kmp/stream.cpp	Sat Dec  7 00:42:19 2002
@@ -31,7 +31,7 @@
 
 #include"ndns.h"
 #include"xmlfilter.h"
-#include"../qssl/qssl.h"
+#include<qssl.h>
 
 using namespace Jabber;
 
