diff -urN psi-0.11.orig/src/applicationinfo.cpp psi-0.11/src/applicationinfo.cpp
--- psi-0.11.orig/src/applicationinfo.cpp	2007-10-14 16:00:51.000000000 +0200
+++ psi-0.11/src/applicationinfo.cpp	2008-02-04 15:09:38.000000000 +0100
@@ -29,7 +29,7 @@
 //#define PROG_VERSION "0.11-dev" " (" __DATE__ ")"; //CVS Builds are dated
 #define PROG_VERSION "0.11";
 #define PROG_CAPS_NODE "http://psi-im.org/caps";
-#define PROG_CAPS_VERSION "0.11-dev-rev8";
+#define PROG_CAPS_VERSION "0.11";
 #define PROG_OPTIONS_NS "http://psi-im.org/options";
 #define PROG_STORAGE_NS "http://psi-im.org/storage";
 
